---
title: "Combined Dada2 Visualization"
author: "Rebecca Clement"
date: "4/2/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
I combined the two seqtab runs into one. Here is the analysis for that.

Load libraries that we need
```{r message=FALSE}
library(dada2)
library(ggplot2)
library(phyloseq)
library(Biostrings)
library(dplyr)
```

```{r}
seqtab_all<-readRDS("~/Box/Clement_TermiteGut_16S_0108/Dada2_rds_files/outfiles/seqtab_both_final.rds")
tax_silva_all<-readRDS("~/Box/Clement_TermiteGut_16S_0108/Dada2_rds_files/outfiles/tax_silva.rds")
tax_dictdb_all<-readRDS("~/Box/Clement_TermiteGut_16S_0108/Dada2_rds_files/outfiles/tax_all_dictdb.rds")
# Read in metadata file
term_meta2<-read.csv("~/Box/Clement_TermiteGut_16S_0108/16S_metadata2.csv")

#add name to seqtab table and export as csv to use in microbiomeanalyst. Rows are OTUs. Samples are columns. Add #NAME to header row
analyst_otu<-t(seqtab_all)
analyst_otu<-rbind(colnames(analyst_otu),analyst_otu)
analyst_otu<-cbind(rownames(analyst_otu),analyst_otu)
analyst_otu[1,1]<-"#NAME"

write.csv(analyst_otu,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/combined_otu.csv",row.names = FALSE, col.names=FALSE)


```
```{r}

```

## 1) Subset our metadata files so that it's equal with the seqtab files.
```{r}
samples_all.out <- rownames(seqtab_all)
samples_all_meta <- term_meta2[term_meta2$Termite_ID %in% samples_all.out,]
rownames(samples_all_meta) <- samples_all_meta$Termite_ID
```
Look at mock communities
```{r}
# Evaluate accuracy on the mock community if sequenced in MiSeq run
```{r}
# mock communities: "BEC0057","flexcleaned","Zymo"
unqs.mock <- seqtab_all["Zymo",]
unqs.mock <- sort(unqs.mock[unqs.mock>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock), "sample sequences present in the Mock community.\n") #92 sample sequences
unqs.mock2 <- seqtab_all["flexcleaned",]
unqs.mock2 <- sort(unqs.mock2[unqs.mock2>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock2), "sample sequences present in the Mock community.\n") #70 sample sequences
unqs.mock3 <- seqtab_all["BEC0057",]
unqs.mock3 <- sort(unqs.mock3[unqs.mock3>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock3), "sample sequences present in the Mock community.\n") #71 sample sequences
unqs.mock4 <- seqtab_all[c("BEC0057","flexcleaned","Zymo"),]
unqs.mock4 <- sort(unqs.mock4[unqs.mock4>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock4), "sample sequences present in the Mock community.\n") #233 sample sequences
write.csv(as.data.frame(unqs.mock),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_zymo.csv")
write.csv(as.data.frame(unqs.mock2),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_flex.csv")
write.csv(as.data.frame(unqs.mock3),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_BEC0057.csv")



mock.ref <- getSequences(file.path(path, "HMP_MOCK.v35.fasta"))
match.ref <- sum(sapply(names(unqs.mock), function(x) any(grepl(x, mock.ref))))
cat("Of those,", sum(match.ref), "were exact matches to the expected reference sequences.\n")

# Conclusion: This mock community contained 20 bacterial strains. DADA2 identified 20 ASVs all of which exactly match the reference genomes of the expected community members. 

```

## 2) Make phyloseq objects from seqtab, metadata and tax. Save these objects.
```{r}
ps_all <- phyloseq(otu_table(seqtab_all, taxa_are_rows=FALSE), 
               sample_data(samples_all_meta), 
               tax_table(tax_silva_all))


# Make the names ASV instead of DNA
dna_all <- Biostrings::DNAStringSet(taxa_names(ps_all))
names(dna_all) <- taxa_names(ps_all)
ps_all <- merge_phyloseq(ps_all, dna_all)
taxa_names(ps_all) <- paste0("ASV", seq(ntaxa(ps_all)))
ps_all
rowSums(otu_table(ps_all))
#write csv of OTU table
write.csv(t(otu_table(ps_all)),"termite_combined_otu.csv") # something seems very wrong with this data
#write taxonomy table
write.csv(tax_table(ps_all),"silva_taxtable.csv")
```

## 3) Compare alpha diversity between samples.
```{r}
plot_richness(ps_all, x="Species", measures=c("Shannon", "Simpson"), color="Site")

```

## 4) Compare beta diversity with ordination. (ex. NMDS)
```{r}
# Transform data to proportions as appropriate for Bray-Curtis distances
ps.prop_all <- transform_sample_counts(ps_all, function(otu) otu/sum(otu))
ord.nmds.bray_all <- ordinate(ps.prop_all, method="NMDS", distance="bray")

plot_ordination(ps.prop_all, ord.nmds.bray_all, color="Species", title="Bray NMDS")

```
```{r}
plot_ordination(ps.prop1, ord.nmds.bray1, color="Site", title="Bray NMDS")

plot_ordination(ps.prop2, ord.nmds.bray2, color="Site", title="Bray NMDS")
```
```{r}
plot_ordination(ps.prop1, ord.nmds.bray1, color="Cluster", title="Bray NMDS")

plot_ordination(ps.prop2, ord.nmds.bray2, color="Cluster", title="Bray NMDS")
```

```{r}
plot_ordination(ps.prop1, ord.nmds.bray1, color="Habitat", title="Bray NMDS")

plot_ordination(ps.prop2, ord.nmds.bray2, color="Habitat", title="Bray NMDS")
```

## 5) Make a bar plot that shows the different samples. 
```{r}
top20_1 <- names(sort(taxa_sums(ps1), decreasing=TRUE))[1:20]
ps.top20_1 <- transform_sample_counts(ps1, function(OTU) OTU/sum(OTU))
ps.top20_1 <- prune_taxa(top20_1, ps.top20_1)
plot_bar(ps.top20_1, x="Species", fill="Family") + facet_wrap(~Species, scales="free_x")

top20_2 <- names(sort(taxa_sums(ps2), decreasing=TRUE))[1:20]
ps.top20_2 <- transform_sample_counts(ps2, function(OTU) OTU/sum(OTU))
ps.top20_2 <- prune_taxa(top20_2, ps.top20_2)
plot_bar(ps.top20_2, x="Species", fill="Family") + facet_wrap(~Species, scales="free_x")
```

