---
title: "Combined Dada2 Visualization"
author: "Rebecca Clement"
date: "4/2/2021"
output: html_document
editor_options: 
  
  This should read in two seqtables, combine them and then run assign taxonomy on them. Updated December 21, 2021
  
  For maps for this project, see the map_making file
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
Load the libraries needed
```{r}
library(dada2) #devtools::install_github("benjjneb/dada2", ref="v1.16")
library(stringr)
library(ggplot2)
library(phyloseq) #BiocManager::install("phyloseq")
library(Biostrings)
library(dplyr)
#devtools::install_github("leffj/mctoolsr")
library(mctoolsr)
library(seqinr)
library(microbiome) #BiocManager::install("microbiome")
library(picante)
library(lme4) #for glms
library(vegan) #for comparing betadiversity
library(DESeq2) #differential abundance BiocManager::install("DESeq2")
library(geodist) #for calculating distance between GPS coordinates
library("reshape2")
library("biomformat") #for picrust2
library("venneuler") #venn diagrams
library(tidyverse)
library(grid)
library(UpSetR) #for making upset plots
library(ComplexHeatmap)
library(scales) #for putting things to percents
library(RColorBrewer)

```
set working directory
```{r}
getwd()
setwd("~/Box/termite_microbiome/")
```
Read in the two seqtabs. skip this if we've already combined
```{r}
seqtab1<-readRDS("../termite_microbiome/Dada2/trimmed_outfiles/seqtab.rds")
seqtab2<-readRDS("../termite_microbiome/Dada2/outfiles/seqtab_set2.rds")

dim(seqtab1) #58x38155 without chimeras. 58x59381 before removing
dim(seqtab2) #73x29757 without chimeras. 73x39879 before removing
```
Combine the two rds files
```{r}
seqtab_all<-mergeSequenceTables(seqtab1,seqtab2)
dim(seqtab_all) #131x88421 , the sum of the two above is 99260. 
seqtab.chim_all <- removeBimeraDenovo(seqtab_all, method="consensus", multithread=TRUE)
dim(seqtab.chim_all) #131x61770
sum(seqtab.chim_all)/sum(seqtab_all) #76.3%
saveRDS(seqtab.chim_all, "~/Box/termite_microbiome/Dada2/final_combined/seqtab.chim_all.rds")
```
Download the most recent version of the SSU Silva database from https://zenodo.org/record/4587955#.YcorTH3MJTY released March 7, 2021
Run assign taxonomy. Note: We did this already on the Mac in the glass classroom, so we can skip this step here.
```{r eval=FALSE, include=FALSE}
seqtab_all<-readRDS("~/Box/termite_microbiome/Dada2/final_combined/seqtab.chim_all.rds")
tax_silva_all<-assignTaxonomy(seqtab_all,"~/Desktop/silva_nr99_v138_wSpecies_train_set.fa.gz",multithread = TRUE)
saveRDS(tax_silva_all, "~/Box/termite_microbiome/Dada2/final_combined/tax_silva_combined.rds")
tax_dictdb_all<-assignTaxonomy(seqtab_all,"~/Desktop/DictDB_DADA2_FINAL.fasta",multithread = TRUE)
saveRDS(tax_dictdb_all, "~/Box/termite_microbiome/Dada2/final_combined/tax_dictdb_combined.rds")
```


Start fresh from here.
```{r}

seqtab_all<-readRDS("~/Box/termite_microbiome/Dada2/Jan2022/seqtab.chim_all.rds")
tax_silva_all<-readRDS("~/Box/termite_microbiome/Dada2/Jan2022/tax_silva_combined2.rds")
tax_dictdb_all<-readRDS("~/Box/termite_microbiome/Dada2/Jan2022/tax_dictdb_combined2.rds")
# Read in metadata file
term_meta2<-read.csv("~/Box/termite_microbiome/16S_metadata2.csv")
```

Prep for microbiome analyst, can skip this
```{r}
#add name to seqtab table and export as csv to use in microbiomeanalyst. Rows are OTUs. Samples are columns. Add #NAME to header row
analyst_otu<-as.data.frame(seqtab_all)
analyst_otu<-t(seqtab_all)
#analyst_otu<-rbind(colnames(analyst_otu),analyst_otu)
analyst_otu_new<-cbind(rownames(analyst_otu),analyst_otu)
colnames(analyst_otu_new)[1]<-"#NAME"
colnames(analyst_otu_new)
#analyst_otu[1,1]<-"#NAME"

#Put this file into both the folders
write.table(analyst_otu_new,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/DictDB_2/combined_otu.csv",sep=",",row.names = FALSE, col.names=T)
write.table(analyst_otu_new,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/Silva_2/combined_otu.csv",sep=",",row.names = FALSE, col.names=T)

# make metadata column name #NAME
analyst_meta<-term_meta2 %>% rename("#NAME" = "Termite_ID")
write.table(analyst_meta,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/meta.csv",sep=",",row.names=F)

# make taxonomy table A1 #TAXONOMY
analyst_tax<-as.data.frame(tax_silva_all) #dimensions: 74755x7
analyst_tax_new<-cbind(rownames(analyst_tax),analyst_tax)
colnames(analyst_tax_new)[1]<-"#TAXONOMY"
write.table(analyst_tax_new,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/Silva_2/combined_silva_tax.csv",sep=",",row.names=F)

analyst_tax_dictdb<-as.data.frame(tax_dictdb_all)
analyst_tax_dictdb_new<-cbind(rownames(analyst_tax_dictdb),analyst_tax_dictdb)
colnames(analyst_tax_dictdb_new)[1]<-"#TAXONOMY"
write.table(analyst_tax_dictdb_new,"~/Box/Clement_TermiteGut_16S_0108/MicrobiomeAnalyst/DictDB_2/combined_dictdb_tax.csv",sep=",",row.names=F)

#Compress the OTU files to a zip file before uploading.

```


## 1) Subset our metadata files so that it's equal with the seqtab files.
```{r}
samples_all.out <- rownames(seqtab_all)
samples_all_meta <- term_meta2[term_meta2$X.NAME %in% samples_all.out,]
rownames(samples_all_meta) <- samples_all_meta$X.NAME
dim(samples_all_meta) #131x13
```
Look at mock communities
```{r}
# Evaluate accuracy on the mock community if sequenced in MiSeq run

# mock communities: "BEC0057","flexcleaned","Zymo"
unqs.mock <- seqtab_all["Zymo",]
unqs.mock <- sort(unqs.mock[unqs.mock>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock), "sample sequences present in the zymo mock community.\n") #DADA2 inferred 91 sample sequences present in the Mock community. Jan2022: 36
unqs.mock2 <- seqtab_all["flexcleaned",]
unqs.mock2 <- sort(unqs.mock2[unqs.mock2>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock2), "sample sequences present in the Mock community.\n") #DADA2 inferred 85 sample sequences present in the Mock community. Jan: 9
unqs.mock3 <- seqtab_all["BEC0057",]
unqs.mock3 <- sort(unqs.mock3[unqs.mock3>0], decreasing=TRUE) # Drop ASVs absent in the Mock
cat("DADA2 inferred", length(unqs.mock3), "sample sequences present in the Mock community.\n") #DADA2 inferred 86 sample sequences present in the Mock community. Jan: 9
#unqs.mock4 <- rbind(unqs.mock,unqs.mock2,unqs.mock3)
#unqs.mock4 <- sort(unqs.mock4[unqs.mock4>0], decreasing=TRUE) # Drop ASVs absent in the Mock
#cat("DADA2 inferred", length(unqs.mock4), "sample sequences present in the Mock community.\n") #233 sample sequences
#write.csv(as.data.frame(unqs.mock),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_zymo.csv")
#write.csv(as.data.frame(unqs.mock2),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_flex.csv")
#write.csv(as.data.frame(unqs.mock3),"~/Box/Clement_TermiteGut_16S_0108/Mock/mock_otu_BEC0057.csv")
```
I downloaded the zymo mock communities from https://s3.amazonaws.com/zymo-files/BioPool/ZymoBIOMICS.STD.refseq.v2.zip, put them into a single fasta file using geneious, and loaded it here.
```{r}
mock.ref <- getSequences(file.path("~/Box/termite_microbiome/bacterial_refs/16S_zymo_all.fasta"))
#Try truncating unqs.mock to taking off last 23.They all end with gctgactgact. 255-23=232. This is a bit of a problem. Not sure if it's big enough to redo everything. It probably is.
#unqs.mock_short<-unqs.mock
#names(unqs.mock_short)<-stringr::str_trunc(names(unqs.mock_short), 232)
match.ref <- sum(sapply(names(unqs.mock), function(x) any(grepl(x, mock.ref))))
cat("Of those,", sum(match.ref), "were exact matches to the expected reference sequences of Zymo.\n") #4? Jan: 9
#unqs.mock_short2<-unqs.mock2
#names(unqs.mock_short2)<-stringr::str_trunc(names(unqs.mock_short2), 232)
match.ref2 <- sum(sapply(names(unqs.mock2), function(x) any(grepl(x, mock.ref))))
cat("Of those,", sum(match.ref2), "were exact matches to the expected reference sequences of flexcleaned.\n") #0? Jan: 9
#unqs.mock_short3<-unqs.mock3
#names(unqs.mock_short3)<-stringr::str_trunc(names(unqs.mock_short3), 232)
match.ref3 <- sum(sapply(names(unqs.mock3), function(x) any(grepl(x, mock.ref))))
cat("Of those,", sum(match.ref3), "were exact matches to the expected reference sequences of BEC057.\n") #1, jan: 9
#match.ref4 <- sum(sapply(names(unqs.mock4), function(x) any(grepl(x, mock.ref))))
#cat("Of those,", sum(match.ref4), "were exact matches to the expected reference sequences of all.\n")

# This didn't really work. Let's try making a plot.
# First make a phyloseq object for just 

```

## 2) Make phyloseq objects from seqtab, metadata and tax. Save these objects.
```{r}
taxa.print <- tax_silva_all # Removing sequence rownames for display only
rownames(taxa.print) <- NULL
head(taxa.print)
table(taxa.print[,1]) # we have 61315 bacteria, 446 Archaea and 2 Eukaryota. Jan: 35284 bac, 165 Archaea, 1 Eukaryote. DictDB: 39089 bacteria only
nrow(taxa.print) #61770 total ASVs. Jan: 35455
nrow(unique(taxa.print)) #799 unique taxa. Jan: 1041 taxa? how are there more with fewer ASVs. DictDB: 144
taxa.print[is.na(taxa.print)] <- "unclassified"
nrow(table(taxa.print[,1])) #3 different kingdoms, 7 unclassified, Jan: 5 unclassified
nrow(table(taxa.print[,2])) #42 different phyla,3859 unclassified, Jan: 46 phyla, 2319 classified
nrow(table(taxa.print[,3])) #88 different classes,6754 unclassified, Jan: 108 classes, 3798 unclassified
nrow(table(taxa.print[,4])) #181 different orders,13386 unclassified, Jan: 218 orders, 7419
nrow(table(taxa.print[,5])) #258 different families,21020 unclassified, Jan: 302 ASV, 14052
nrow(table(taxa.print[,6])) #379 different genera, 33299 unclassified at genus level, Jan: 495, 21602 unclassified
nrow(table(taxa.print[,7])) #170 different species, Jan: 237 species, 34764


table(taxa.print[,1]) # There are 7 unclassified
tax_silva_all[which(taxa.print[,1]=="Eukaryota"),] # a blast shows that one of these is Cryptococcus neoformans, a fungus and the other is uncultured leafcutter ant microbe
nrow(taxa.print[which(taxa.print[,4]=="Chloroplast"),]) # There are 14 Chloroplast, Jan: 12
nrow(taxa.print[which(taxa.print[,5]=="Mitochondria"),]) # There are 16 that have family Mitochondria, Jan: 80
tax_silva_all[is.na(tax_silva_all)] <- "unclassified"  # change NA to unclasssified
sort(unique(tax_silva_all[,1]))
# Get rid of Chloroplast and mitochondria
taxa2<-data.frame(tax_silva_all) ## convert to data.frame
taxa2<-subset(taxa2, Order!="Chloroplast" & Family!="Mitochondria" & Kingdom!="unclassified")  # I keep bacteria and eliminate chloroplasts and mitochondria




b<-t(seqtab_all) # transform it
ASVsfiltered<-b[rownames(taxa2),]
seqtab_all2<-t(ASVsfiltered);rm(b);rm(ASVsfiltered) #keep only those with filtered taxonomy
dim(seqtab_all2) # 131x61733, Jan: 35358
dim(taxa2) #61733x7
# change long names in reads to amplicon single variants (ASV)
colnames(seqtab_all2) <- paste0("ASV", 1:ncol(seqtab_all2))
taxa3 <- taxa2
rownames(taxa3) <- paste0("ASV", 1:nrow(taxa3))
write.table(taxa3,"~/Box/termite_microbiome/Dada2/Jan2022/taxa_asvnames.txt",quote=F,sep="\t")

## export tables with ASVs
# run function
export_taxa_table_and_seqs = function(seqtab, file_seqtab, file_seqs) {
  seqtab.t = as.data.frame(t(seqtab))
  seqs = row.names(seqtab.t)
  row.names(seqtab.t) = paste0("ASV", 1:nrow(seqtab.t))
  outlist = list(data_loaded = seqtab.t)
  mctoolsr::export_taxa_table(outlist, file_seqtab)
  seqs = as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab.t), file_seqs)
}

export_taxa_table_and_seqs(seqtab_all,"~/Box/termite_microbiome/Dada2/Jan2022/ASV_table_test_conc.txt","~/Box/termite_microbiome/Dada2/Jan2022/ASV_seqs_test_conc.fa")


tree <- read_tree("~/Box/termite_microbiome/Dada2/Jan2022/ASV_treefile_Jan2022_newick")
#remove everything before the underscore in tip labels
tree$tip.label<-gsub(".*_","",tree$tip.label)
taxa1 <- read.delim("~/Box/termite_microbiome/Dada2/final_combined/taxa_asvnames.txt")
otu <- read.delim("~/Box/termite_microbiome/Dada2/final_combined/ASV_table_test_conc.txt", skip = 1, row.names = 1, check.names = FALSE)
metadata <- samples_all_meta
#Change names to be less ambiguous with taxonomy
colnames(metadata)[c(4,10)]<-c("t_sp","t_fam")
ps <- phyloseq(tax_table(as.matrix(taxa1)), phy_tree(tree), sample_data(metadata), otu_table(otu, taxa_are_rows = TRUE))
ps1 <- ps

# if you don't have a tree
ps_all <- phyloseq( tax_table(as.matrix(taxa1)), otu_table(otu, taxa_are_rows=TRUE), 
               sample_data(metadata))
# Make the names ASV instead of DNA
#dna_all <- Biostrings::DNAStringSet(taxa_names(ps_all))
#names(dna_all) <- taxa_names(ps_all)
#ps_all <- merge_phyloseq(ps_all, dna_all)
#taxa_names(ps_all) <- paste0("ASV", seq(ntaxa(ps_all)))
#ps_all #73K taxa, 128 samples
#rowSums(otu_table(ps_all))
#mean(rowSums(otu_table(ps_all))) #148591.5 reads/sample
#write csv of OTU table
#write.csv(t(otu_table(ps_all)),"termite_combined_otu.csv") # something seems very wrong with this data
#write taxonomy table
#write.csv(tax_table(ps_all),"silva_taxtable.csv")
```
2.5 Summarize taxa through plots
```{r}
theme_set(theme_bw())
#remove extra taxonomic ranks
# remove ASVs whose mean value per sample is less than 1

ps3 <- filter_taxa(ps1, function(x) mean(x) > 1, TRUE)
nrow(ps1@otu_table) #61733, Jan: 35455
nrow(ps3@otu_table) #20159, Jan: 20159

# fiter singletons
ps4 <- prune_taxa(taxa_sums(ps3) > 1, ps3) 
nrow(ps4@otu_table) #20159, no change from previous
# filter samples with less than 1000 reads
ps5 = prune_samples(sample_sums(ps4) >= 1000, ps4)
nrow(ps5@otu_table) #20159, no change
#without zymo


#Export table of ASVs
table_otu<-otu_table(ps5)
write.csv(table_otu,file="~/Box/termite_microbiome/Dada2/Jan2022/silva_table_ps5_test_conc.csv")

#Export taxonomy of ASVs
table_tax<-tax_table(ps5)
write.csv(table_tax,file="~/Box/termite_microbiome/Dada2/Jan2022/silva_tax_ps5_test_conc.csv")

# Export table of ASVs with taxonomy
table_all<-cbind(tax_table(ps5),otu_table(ps5))
write.csv(table_all,file="~/Box/termite_microbiome/Dada2/Jan2022/table_tax_ps5.csv")
```


2.75 make plots'
Make a plot. x axis is ordered by species
```{r}
seq.results <- read.csv("~/Box/termite_microbiome/n_reads_track.csv")
#put into long format
long_seq.results<-melt(data=seq.results,
                       id.vars=c("X","Sequencing_group","t_sp","Site"),
                       variable.name="Status",
                       value.name="Reads")

long_seq.results$Status <- factor(long_seq.results$Status, levels=c("input","filtered","denoisedF","denoisedR","merged","nonchim")) #this makes raw first
long_seq.results %>%
  group_by(Sequencing_group,t_sp)%>%
  summarise(mean=mean(Reads))
#remove unwanted levels
seq.results_sub<-filter(long_seq.results, Status=="input"|Status=="nonchim")
seq.results_sub<-filter(seq.results_sub, t_sp!="zymo")

seq.results_plot <- ggplot(data=seq.results_sub, aes(x=X, y=Reads, fill=Status)) + geom_bar(stat="identity", position="dodge")
seq.results_plot <- seq.results_plot + theme_classic() + theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0)) +facet_grid(.~t_sp+ Sequencing_group, scales = "free", space = "free")
seq.results_plot
ggsave("~/Box/termite_microbiome/Figures/seq.results_plot.pdf", seq.results_plot,width=10)
```


## 3) Compare alpha diversity between samples.
```{r}
ps6<-subset_samples(ps5,t_sp!="zymo")
summarize_phyloseq(ps6) #tells min: 27950, max:281879, total:14437993, average:112797, median:88753 #singletons: 72 etc.

# change legend name
#colnames(ps6@sam_data$Site1)  <- "Collection site"
rich_site<-plot_richness(ps6, x="Site1", measures=c("Shannon"), color="Site1")+facet_wrap(~t_sp1, scales="free_x")+theme_bw()+ theme(axis.text.x = element_blank())+ scale_color_brewer(palette="Dark2")+xlab('Site1')+ylab('Shannon Diversity')
rich_site

ggsave("~/Box/termite_microbiome/Dada2/Jan2022/richness2.png",rich_site)

```


Run code from dada2_marcos.R
Data normalization
This will use a negative binomial distribution
```{r}
diagdds = phyloseq_to_deseq2(ps6, ~t_sp) # make a deseq object
# Calculate geometric means
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
# Estimate size factors
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
# Get Normalized read counts
normcounts <- counts(diagdds, normalized = TRUE)
# Round read counts
normcountsrd<-round(normcounts, digits = 0)
# Transform matrix of normalized counts to phyloseq object
ncr<-otu_table(normcountsrd, taxa_are_rows = TRUE) 
# Replace otu_table in original phyloseq object
ps6b<-ps6
otu_table(ps6b) <- ncr
ps6<-ps6b
write.csv(ncr,file="~/Box/termite_microbiome/Dada2/Jan2022/deseq_file_normalized.csv")
```

Reorder variables
```{r}
# Change factor levels to be in different order
#ps6<-subset_samples(ps6,t_sp!="zymo")
ps6@sam_data$t_sp1 <- as.factor(ps6@sam_data$t_sp)
levels(ps6@sam_data$t_sp1)<-c('Amitermes','Coptotermes','Microcerotermes','Microcerotermes','Nasutitermes')
ps6@sam_data$t_sp1<-factor(ps6@sam_data$t_sp1, levels=c('Amitermes','Microcerotermes','Nasutitermes','Coptotermes'))

# Change sites to be driest to wettest
ps6@sam_data$Site1<-as.factor(ps6@sam_data$Site)
levels(ps6@sam_data$Site1)<- c('JCU rainforest','Sclerophyll','Dry savanna','Restored rainforest','Wet savanna')
ps6@sam_data$Site1<-factor(ps6@sam_data$Site1,levels=c('Dry savanna','Wet savanna','Sclerophyll','JCU rainforest','Restored rainforest'))
ps6@sam_data$Site2<-ps6@sam_data$Site1
levels(ps6@sam_data$Site2)<- c('Sav1','Sav2','Scl1','RftJ','RftR')


#Change Microcerotermes serratus to sp G
#levels(ps6@sam_data$t_sp)<-c(levels(ps6@sam_data$t_sp),"M_sp_G")
ps6@sam_data$t_sp[100:128]<-"M_sp_G"
ps6@sam_data$t_sp<-factor(ps6@sam_data$t_sp,levels=c("A_laurensis", "M_serratus","M_sp_G","N_magnus","C_acinaciformis"))
```
## 3) Compare alpha diversity between samples after normalization.
```{r}
# change legend name
#colnames(ps6@sam_data$Site1)  <- "Collection site"
rich_site<-plot_richness(ps6, x="Site1", measures=c("Shannon"), color="Site1")+facet_wrap(~t_sp1, scales="free_x")+theme_bw()+ theme(axis.text.x = element_blank())+ scale_color_brewer(palette="Dark2")+xlab('Site1')+ylab('Shannon Diversity')
rich_site

rich_site_genus<-plot_richness(ps.bygenus, x="Site1", measures=c("Shannon"), color="Site1")+facet_wrap(~t_sp1, scales="free_x")+theme_bw()+ theme(axis.text.x = element_blank())+ scale_color_brewer(palette="Dark2")+xlab('Site1')+ylab('Genus Shannon Diversity')
rich_site_genus

#Fisher is theoretically independent of sample size, so maybe we should use this metric instead
rich_site_fish<-plot_richness(ps6, x="Site1", measures=c("Fisher"), color="Site1")+facet_wrap(~t_sp1, scales="free_x")+theme_bw()+ theme(axis.text.x = element_blank())+ scale_color_brewer(palette="Dark2")+xlab('Site1')+ylab('Fisher Diversity')
rich_site_fish

ggsave("~/Box/termite_microbiome/Dada2/Jan2022/richness_normalized.png",rich_site)

# Chao1 estimates diversity from abundance data
rich_site_chao1<-plot_richness(ps6, x="Site1", measures=c("Chao1"), color="Site1")+facet_wrap(~t_sp1, scales="free_x")+theme_bw()+ theme(axis.text.x = element_blank())+ scale_color_brewer(palette="Dark2")+xlab('Site1')+ylab('Chao1 Diversity')
rich_site_chao1

ggsave("~/Box/termite_microbiome/Dada2/Jan2022/chao_richness_normalized.png",rich_site)

one.way <- aov(value ~ Site1, data = rich_site$data[rich_site$data$t_sp=="M_serratus",])
#shannon diversity is a combination of species richness and evenness
summary(one.way) # no significant difference in diversity of just M_serratus
one.way2 <- aov(value ~ Site1, data = rich_site$data[rich_site$data$Site=="PW"|rich_site$data$Site=="SC",])
summary(one.way2) #1.18e-06 significant difference
TukeyHSD(one.way2) 
confint(one.way2) #95% confident that wet savanna is higher in shannoon diversity by .24-.53
t.test(rich_site$data$Site=="PW",rich_site$data$Site=="SC") #mean of PW: 0.3162393 Mean of SC: 0.4444444 p=.0436
```


Analysis of all samples
```{r}
otuD<-as.data.frame(t(otu_table(ps6)))
phylodiversityNORM_Q<-pd(otuD, phy_tree(ps6), include.root=TRUE) ### phyilogenetic diversity. Include root =True tree rooted via midpoint
diversityNORM_Q<-estimate_richness(ps6)
diversityNORM_Q1<-cbind(sample_data(ps6),diversityNORM_Q,phylodiversityNORM_Q)
# Export selected alpha-diversity indices
write.csv(diversityNORM_Q1,file="~/Box/termite_microbiome/Dada2/Jan2022/alpha_div_normalized.csv")

# Make a diversity df for genus level
otuD<-as.data.frame(t(otu_table(ps.bygenus)))
phylodiversityNORM_genus<-pd(otuD, phy_tree(ps.bygenus), include.root=TRUE) 
diversityNorm_genus<-estimate_richness(ps.bygenus)
diversityNORM_genus<-cbind(sample_data(ps.bygenus),diversityNorm_genus,phylodiversityNORM_genus)
```

Multiplot function
```{r}
par(mfrow = c(2, 2))

######## funcion multiplot ######## 

multiplot <- function(..., plotlist=NULL, file, cols=1, layout=NULL) {
  library(grid)
  
  # Make a list from the ... arguments and plotlist
  plots <- c(list(...), plotlist)
  
  numPlots = length(plots)
  
  # If layout is NULL, then use 'cols' to determine layout
  if (is.null(layout)) {
    # Make the panel
    # ncol: Number of columns of plots
    # nrow: Number of rows needed, calculated from # of cols
    layout <- matrix(seq(1, cols * ceiling(numPlots/cols)),
                     ncol = cols, nrow = ceiling(numPlots/cols))
  }
  
  if (numPlots==1) {
    print(plots[[1]])
    
  } else {
    # Set up the page
    grid.newpage()
    pushViewport(viewport(layout = grid.layout(nrow(layout), ncol(layout))))
    
    # Make each plot, in the correct location
    for (i in 1:numPlots) {
      # Get the i,j matrix positions of the regions that contain this subplot
      matchidx <- as.data.frame(which(layout == i, arr.ind = TRUE))
      
      print(plots[[i]], vp = viewport(layout.pos.row = matchidx$row,
                                      layout.pos.col = matchidx$col))
    }
  }
}
```

Boxplots for diversity
```{r}
# We should really only do these with Sav1 & Sav2
s1s2div<-diversityNORM_Q1[1:99,]
observed <- ggplot(s1s2div, aes(factor(t_sp), Observed))
observed2<-observed + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1,show.legend = FALSE)+ geom_jitter(size=1,shape=1)+labs(y = "ASV richness",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
chao <- ggplot(diversityNORM_Q1, aes(factor(t_sp1), Chao1))
chao2<-chao + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Chao1 richness",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
chao2 #nonparameteric asymptotic estimator of species richness
shan <- ggplot(s1s2div, aes(factor(t_sp), Shannon))
shan2<-shan + geom_boxplot(aes(fill = Site1), outlier.size = 1,show.legend = FALSE)+ geom_jitter(size=1,shape=1)+labs(y = "ASV Shannon diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
#Shannon diversity index (H): Includes species richness and species evenness--more weight on richness.
#Simpson diversity index (D): Estimator of richness and evenness--more weight on species evenness.
invsimp <- ggplot(diversityNORM_Q1, aes(factor(t_sp1), InvSimpson))
invsimp2<- invsimp + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+ labs(y = "InvSimpson diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
invsimp2 #effective number of types that is obtained when the weighted arithmetic mean is used to quantify average proportional abundance of types in the dataset of interest.
ACE <- ggplot(diversityNORM_Q1, aes(factor(t_sp1), ACE))
ACE2<- ACE + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+ labs(y = "ACE diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
ACE2 #Involves an arbitrary abundance threshold. Only abundance of abundant species is considered.
phyl <- ggplot(diversityNORM_Q1, aes(factor(t_sp1), PD))
phyl2<-phyl + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Faith's diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
fish <- ggplot(diversityNORM_Q1, aes(factor(t_sp1), Fisher))
fish2<- fish + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Fisher diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
multiplot(observed2, shan2, chao2, fish2, cols=2)  #### to run this fucntion I have to run function multiplot first
multiplot(fish2, ACE2, cols=2)  #### to run this fucntion I have to run function multiplot first

#Diversity of genera
s1s2div_genus<-diversityNORM_genus[1:90,]
gen_observed<- ggplot(s1s2div_genus, aes(factor(t_sp), Observed))
gen_observed2<-gen_observed + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Generic richness",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
gen_shan <- ggplot(s1s2div_genus, aes(factor(t_sp), Shannon))
gen_shan2<-gen_shan + geom_boxplot(aes(fill = Site1),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Generic Shannon diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")+theme_classic()
multiplot(observed2,shan2,gen_observed2,gen_shan2,cols=2)
```

Alpha diversity comparisons
```{r}
#Kruskal wallace between sequencing batches
kruskal.test(diversityNORM_Q1$Observed,diversityNORM_Q1$Sequencing_group) #yes significant p=2.7e-16
kruskal.test(diversityNORM_Q1$Shannon,diversityNORM_Q1$Sequencing_group)
kruskal.test(diversityNORM_Q1$Chao1,diversityNORM_Q1$Sequencing_group)
kruskal.test(diversityNORM_Q1$Fisher,diversityNORM_Q1$Sequencing_group)
# We should really only do these with Sav1 & Sav2
s1s2div<-diversityNORM_Q1[1:99,]
table(s1s2div$t_sp)
t1<-lmer(Observed~t_sp1+Site1+ (1|Sequencing_group), s1s2div)
summary(t1) 
confint(t1) #Serratus is more likely to be 124-392 higher. MS and RR are higher than


t1_shan<-lmer(Shannon~t_sp1+Site1+ (1|Sequencing_group), s1s2div)
summary(t1_shan) 
confint(t1_shan)
t1_fish<-lmer(Fisher~t_sp1+Site+Habitat+Feeding_Group+ (1|Sequencing_group), diversityNORM_Q1)

#Genus comparisons
s1s2div_genus<-diversityNORM_genus[1:90,]
t1_genus<-lmer(Observed~t_sp1+Site1+ (1|Sequencing_group), s1s2div_genus)
summary(t1_genus)
confint(t1_genus)
t1_shan_genus<-lmer(Shannon~t_sp1+Site1+ (1|Sequencing_group), s1s2div_genus)
summary(t1_shan_genus) 
confint(t1_shan_genus)
anova(t1_shan_genus,t1_genus)

#Use fisher diversity which is theoretically independent of sample size.
t2<-lm(Chao1~Site, diversityNORM_Q1[diversityNORM_Q1$t_sp1=="Microcerotermes",])
t2
confint(t2)

```
Calculate some summary statistics
```{r}
head(diversityNORM_Q1)

s1s2div_Q1 %>% 
  group_by(t_sp1) %>% 
  summarise_at(vars(Observed,Shannon),list(name=mean))
s1s2div_Q1 %>% 
  group_by(Site1) %>% 
  summarise_at(vars(Observed,Shannon),list(name=mean))
s1s2div_genus %>% 
  group_by(t_sp1) %>% 
  summarise_at(vars(Observed,Shannon),list(name=mean))
s1s2div_genus %>% 
  group_by(Site1) %>% 
  summarise_at(vars(Observed,Shannon),list(name=mean))
```


#Try these on just the dataset with Microcerotermes
```{r}
microdiv<-diversityNORM_Q1[diversityNORM_Q1$t_sp1=="Microcerotermes",]
levels(microdiv$t_sp)<-c("A_laurensis","C_acinaciformis","M_serratus","M_sp_G","N_magnus")
microdiv$t_sp[-(1:13)]<-"M_sp_G"
t2<-lm(Fisher~`Collection site`+Habitat, microdiv)
summary(t2) # significant difference JCU is lower than dry savanna
shan <- ggplot(microdiv, aes(factor(t_sp), Shannon))
shan2<-shan + geom_boxplot(aes(fill = factor(Site1)),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Shannon diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+theme_classic()
shan2
observed <- ggplot(microdiv, aes(factor(t_sp), Observed))
observed2<-observed + geom_boxplot(aes(fill = factor(Site1)),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "ASV richness",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+theme_classic()
observed2
chao <- ggplot(microdiv, aes(factor(t_sp), Chao1))
chao2<-chao + geom_boxplot(aes(fill = factor(Site1)),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Chao1 richness",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
chao2
fish <- ggplot(microdiv, aes(factor(t_sp), Fisher))
fish2<- fish + geom_boxplot(aes(fill = factor(Site1)),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Fisher diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))
fish2
phyl <- ggplot(microdiv, aes(factor(t_sp), PD))
phyl2<-phyl + geom_boxplot(aes(fill = factor(Site1)),outlier.colour = "black", outlier.size = 1)+ geom_jitter(size=1,shape=1)+labs(y = "Faith's diversity",x="")+ theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust=1))+ scale_color_brewer(palette="Dark2")
phyl2
multiplot(observed2,shan2,chao2,fish2,cols=2)
multiplot(observed2,shan2,cols=2)

t1_m<-lm(Observed~Site1, microdiv)
summary(t1_m)
t1_m2<-lm(Observed~t_sp, microdiv)
summary(t1_m2)
t1_m_shan<-lm(Shannon~Site1, microdiv)
summary(t1_m_shan)
t1_m_shan2<-lm(Shannon~t_sp, microdiv)
summary(t1_m_shan2)

```
Filter out samples that don't seem to fit
```{r}
ord.nmds.bray_ps5<- ordinate(ps5, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps5, ord.nmds.bray_ps5, color="t_sp", title="Bray NMDS PS5",label="X.NAME") # to see before normalization


ord.nmds.bray_ps6<- ordinate(ps6, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps6, ord.nmds.bray_ps6, color="t_sp", title="Bray NMDS PS6",label="X.NAME") #after normalization
ps7<-ps6
# BEC321 (low read count), BEC270 (low count), BEC292 (low read count), ok: BEC278 (low read count)
# 4 and 10 are misidentified and 23. Actually nasutitermes
ps7@sam_data[c("BEC0004","BEC0010","BEC0023"),c("t_sp","t_sp1","t_fam","Subfamily")]<-ps7@sam_data[c("BEC0003"),c("t_sp","t_sp1","t_fam","Subfamily")]
# 270 should be Amitermes
ps7@sam_data[c("BEC270"),c("t_sp","t_sp1","t_fam","Subfamily")]<-ps7@sam_data[c("BEC0002"),c("t_sp","t_sp1","t_fam","Subfamily")]
colSums(ps7@otu_table) #read counts
# Which ones have the lowest read counts?
colSums(ps7@otu_table[, order(colSums(ps6@otu_table))])
# ones that are less than 57K and 28 bc I don't like it
lowcounts<-c("BEC321","BEC278","BEC268","BEC0018","BEC280","BEC292","BEC0028", "BEC256","BEC0031","BEC274","Zymo","BEC0057","flexcleaned")
# 263,253,257,254,255,"BEC270"
# other problematic ones
problematic<-c("BEC0004","BEC0010","BEC0023","BEC0015","BEC0028")
 #make a thing that has the ones filtered out
ps7<-subset_samples(ps7,!X.NAME %in% lowcounts)
ord.nmds.bray_ps7<- ordinate(ps7, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps7, ord.nmds.bray_ps7, color="Site1",shape="t_sp", title="Bray NMDS")#,label="X.NAME")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/NMDS_ps7_Bray.png")
ord.nmds.bray_ps7<- ordinate(ps7, method="NMDS", distance="jaccard")
phyloseq::plot_ordination(ps7, ord.nmds.bray_ps7, color="Site1",shape="t_sp", title="Jaccard NMDS")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/NMDS_ps7_Jaccard.png")
ord.nmds.bray_ps7<- ordinate(ps7, method="NMDS", distance="unifrac")
phyloseq::plot_ordination(ps7, ord.nmds.bray_ps7, color="Site1",shape="t_sp", title="Unifrac NMDS")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/NMDS_ps7_Unifrac.png") # Is it possible that unifrac shows that Amitermes and M. serratus are more phylogenetically similar?

ps_set1<-subset_samples(ps7,Sequencing_group=="B1")
#ps_set1_g<-subset_samples(ps_set1,!X.NAME %in% lowcounts)
ps_set2<-subset_samples(ps7,Sequencing_group=="B2")
#ps_set2_g<-subset_samples(ps_set2,!X.NAME %in% lowcounts)
ord.nmds.bray_all_set1 <- ordinate(ps_set1, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps_set1, ord.nmds.bray_all_set1, color="Site1", shape="t_sp", title="Bray NMDS",label="X.NAME")
# outliers from species: BECC0031 (low read count),BEC004 & BEC010 (same mound, probably mis-identification), BEC0023 (probably mis-ID), OK: BEC0015 (low-ish count),BEC0028 (low count),BEC0018 (low read count)
ord.nmds.bray_all_set2 <- ordinate(ps_set2, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps_set2, ord.nmds.bray_all_set2, color="Site1", shape="t_sp", title="Bray NMDS",label="X.NAME")



```
Make a phyloseq object of just the genus level
```{r}
ps.bygenus<-tax_glom(ps7,taxrank='Genus')
ps.bygenus <- prune_taxa(taxa_sums(ps.bygenus)>0, ps.bygenus)
nrow(ps.bygenus@otu_table) #280, 277 before pruning, 264 after trimming
length(unique(ps7@tax_table[,"Species"])) #3 kingdoms,29 phyla,54 classes,88 order,119 families,131 genus, 56 species. Jan: 2 kingdoms, 29 phyla, 55 classes, 92 orders, 123 Families, 130 genera, 46 species
length(unique(ps.bygenus@tax_table[,"Species"])) #same as above but 1 species

```

```{r}
# Transform to percentages
ps.tr <- transform_sample_counts(ps7, function (x) x / sum(x))
ps.bygenus.tr <- transform_sample_counts(ps.bygenus, function (x) x / sum(x))
```
Genus ordination
```{r}
#try genus ordination

ord.nmds.bray_genus <- ordinate(ps.bygenus.tr, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps.bygenus.tr, ord.nmds.bray_genus, color="Site1", shape="t_sp", title="Genus Bray NMDS")

phyloseq::plot_ordination(ps.bygenus.tr, ord.nmds.bray_genus, shape="Site2",color="Sequencing_group", title="Bray NMDS")

ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/NMDS_site_genus.png", NMDS_site_genus)

# Using unifrac distance
ord.nmds.unifrac_genus <- ordinate(ps.bygenus.tr, method="NMDS", distance="unifrac")
phyloseq::plot_ordination(ps.bygenus.tr, ord.nmds.unifrac_genus, color="Site1", shape="t_sp", title="Unifrac NMDS")

# transformed ordination
ord.nmds.bray_tr <- ordinate(ps.tr, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps.tr, ord.nmds.bray_tr, shape="t_sp", color="Site1", title="Transformed Bray NMDS")+ labs(shape='Termite species',color="Collection site") 
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/NMDS_site_transformed.png")
```
make plot by genus
```{r}
ps.bygenus.tr.f <- filter_taxa(ps.bygenus.tr, function (x) mean(x) > .005, TRUE) 
patho.by_genus_plot <- plot_bar(ps.bygenus.tr.f, "Sample", "Abundance", 'Genus', labs(y="Relative Abundance", title = "Genus-level assignments"))
ps.by_genus_plot <- patho.by_genus_plot + facet_grid(.~ t_sp1+Site2, scales = "free", space = "free") +
  #geom_rect(aes(fill=Site1),xmin =-Inf,xmax=Inf,ymin=-Inf,ymax=Inf,alpha = 0.1)+
  theme_classic() + 
  #facet_wrap(~t_sp, scales="free_x")+
  #facet_grid( .~ t_sp, scales = "free", space = "free") + 
  theme(legend.key.size = unit(.2, "cm")) + 
  guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
  theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
  scale_fill_discrete(name="Genus")+
  ggtitle("")
ps.by_genus_plot
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/MPS.by_genus_0.005.pdf", ps.by_genus_plot)


```


make a function for plot
```{r}
plot_bar_plot <- function(ps.any, ., Site1,keep_percent) {
  ps.any.tr <- transform_sample_counts(ps.any, function (x) x / sum(x))
ps.any.f <- filter_taxa(ps.any.tr, function (x) mean(x) > keep_percent, TRUE) 
  patho.by_any_plot <- plot_bar(ps.any.f, "Sample", "Abundance", 'Genus', labs(y="Relative Abundance", title = "Genus-level assignments"))
  ps.by_any_plot <- patho.by_any_plot + facet_grid( .~ Site1, scales = "free", space = "free") +
    theme_classic() + 
    #facet_wrap(~t_sp, scales="free_x")+
    #facet_grid( .~ t_sp, scales = "free", space = "free") + 
    theme(legend.key.size = unit(.2, "cm")) + 
    guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
    scale_fill_discrete(name="Genus")+
    ggtitle("")
  ps.by_any_plot 
}
```
Plot phylum
```{r}
ps.any.tr <- transform_sample_counts(ps.byphylum, function (x) x / sum(x))
keep_percent<-.005
ps.any.f <- filter_taxa(ps.any.tr, function (x) mean(x) > keep_percent, TRUE) 
  patho.by_any_plot <- plot_bar(ps.any.f, "Sample", "Abundance", 'Phylum', labs(y="Relative Abundance", title = "Phylum-level assignments"))
  ps.by_phylum_plot <- patho.by_any_plot + facet_grid( .~ t_sp, scales = "free", space = "free") +
    theme_classic() + 
    #facet_wrap(~t_sp, scales="free_x")+
    #facet_grid( .~ t_sp, scales = "free", space = "free") + 
    theme(legend.key.size = unit(.2, "cm")) + 
    guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
    scale_fill_discrete(name="t_sp")+
    ggtitle("")
  ps.by_phylum_plot 
  ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/MPS.by_phylum_0.005.pdf", ps.by_phylum_plot)
```

Make one of these for each species looking at difference in site
```{r}
nrow(ps7@otu_table) #20159 
nrow(ps.bygenus@otu_table) #277

# How many taxa are there?
ps8 <- prune_taxa(taxa_sums(ps7)>0, ps7)
nrow(ps8@otu_table) #18591, #Jan 19017


m_serratus<-subset_samples(ps.bygenus,t_sp1=="Microcerotermes")
m_serratus<-prune_taxa(taxa_sums(m_serratus)>0, m_serratus)
m_serratus1<-subset_samples(ps.tr,t_sp1=="Microcerotermes") # with transformed data
m_serratus1<-prune_taxa(taxa_sums(m_serratus1)>0, m_serratus1)
m_serratus2<-subset_samples(ps8,t_sp1=="Microcerotermes") #with whole data
m_serratus2<-prune_taxa(taxa_sums(m_serratus2)>0, m_serratus2)
nrow(m_serratus@otu_table)
nrow(m_serratus1@otu_table)#4448 OTUs, 109 genera, Jan: 109
nrow(m_serratus2@otu_table)

a_laurensis<-subset_samples(ps.bygenus,t_sp=="A_laurensis")
a_laurensis<-prune_taxa(taxa_sums(a_laurensis)>0, a_laurensis)
a_laurensis1<-subset_samples(ps.tr,t_sp=="A_laurensis")
a_laurensis1<-prune_taxa(taxa_sums(a_laurensis1)>0, a_laurensis1)
nrow(a_laurensis@otu_table) #7652 OTUs, 184 genera
nrow(a_laurensis1@otu_table)
a_laurensis2<-subset_samples(ps8,t_sp=="A_laurensis")
a_laurensis2<-prune_taxa(taxa_sums(a_laurensis2)>0, a_laurensis2)

c_acinaciformis<-subset_samples(ps.bygenus,t_sp=="C_acinaciformis")
c_acinaciformis<-prune_taxa(taxa_sums(c_acinaciformis)>0, c_acinaciformis)
c_acinaciformis1<-subset_samples(ps.tr,t_sp=="C_acinaciformis")
c_acinaciformis1<-prune_taxa(taxa_sums(c_acinaciformis1)>0, c_acinaciformis1)
nrow(c_acinaciformis@otu_table) # 4069 OTUs, 145 genera, Jan: 135
nrow(c_acinaciformis1@otu_table) 
c_acinaciformis2<-subset_samples(ps8,t_sp=="C_acinaciformis")
c_acinaciformis2<-prune_taxa(taxa_sums(c_acinaciformis2)>0, c_acinaciformis2)

n_magnus<-subset_samples(ps.bygenus,t_sp=="N_magnus")
n_magnus<-prune_taxa(taxa_sums(n_magnus)>0, n_magnus)
n_magnus1<-subset_samples(ps.tr,t_sp=="N_magnus")
n_magnus1<-prune_taxa(taxa_sums(n_magnus1)>0, n_magnus1)
nrow(n_magnus@otu_table) #4534 OTUs, 155 genera, 
nrow(n_magnus1@otu_table)
n_magnus2<-subset_samples(ps8,t_sp=="N_magnus")
n_magnus2<-prune_taxa(taxa_sums(n_magnus2)>0, n_magnus2)

zymo<-subset_samples(ps5,t_sp=="zymo")
zymo<-prune_taxa(taxa_sums(zymo)>0, zymo)
zymo1<-tax_glom(zymo,taxrank='Genus')
nrow(zymo@otu_table) #85 OTUS, 9 genera, Jan: 85 OTUS
nrow(zymo1@otu_table) #Jan: 9 genera
zymo.tr<-transform_sample_counts(zymo, function(otu) otu/sum(otu))
zymo1.tr<-transform_sample_counts(zymo1, function(otu) otu/sum(otu))

#Subspecies of Microcerotermes
Mserratus<-subset_samples(m_serratus,t_sp=="M_serratus")
MspG<-subset_samples(m_serratus,t_sp=="M_sp_G")
Mserratus1<-subset_samples(m_serratus1,t_sp=="M_serratus")
MspG1<-subset_samples(m_serratus1,t_sp=="M_sp_G")
Mserratus2<-subset_samples(m_serratus2,t_sp=="M_serratus")
MspG2<-subset_samples(m_serratus2,t_sp=="M_sp_G")


# Microcerotermes
ord.nmds.bray_micro <- ordinate(m_serratus, method="NMDS", distance="bray")
m_serratus_plot<-phyloseq::plot_ordination(m_serratus, ord.nmds.bray_micro, color="Site1",shape="t_sp", title="Microcerotermes Bray NMDS genus") # at the genus level
m_serratus_plot<-m_serratus_plot + stat_ellipse(type="norm", linetype=1) +  theme_bw()
m_serratus_plot
#ordiellipse(ord.nmds.bray_micro, groups = "species", display = "sites", kind = "sd", conf = 0.95, col = "black", w = NULL)
#nmds_phyloseq(m_serratus, treatment, method = 'bray', circle = 0.95,
#labels = NULL, colors = 'default', verbose = TRUE)
#try with transformed data
ord.nmds.bray_micro2 <- ordinate(m_serratus2, method="NMDS", distance="bray")
phyloseq::plot_ordination(m_serratus2, ord.nmds.bray_micro2, color="Site1", title="Microcerotermes Bray NMDS ps7") #with transformed data
ord.nmds.unifrac_micro2 <- ordinate(m_serratus2, method="NMDS", distance="unifrac")
phyloseq::plot_ordination(m_serratus2, ord.nmds.unifrac_micro2, color="Site1", title="Microcerotermes Unifrac NMDS ps7") #with transformed data

ord.nmds.bray_micro.tr <- ordinate(m_serratus1, method="NMDS", distance="bray")
phyloseq::plot_ordination(m_serratus1, ord.nmds.bray_micro.tr, color="Site1", title="Microcerotermes Bray NMDS transformed")+ stat_ellipse(type="norm", linetype=1) +  theme_bw() #with ASV level data
plot_bar_plot(m_serratus,.,Site1,0.005)
# Higher levels of ruminococcus (breaks down cellulose) in SCL and JCU
# M_serratus vs M_sp_G
ord.nmds.bray_micro_s <- ordinate(Mserratus1, method="NMDS", distance="bray")
phyloseq::plot_ordination(Mserratus1, ord.nmds.bray_micro_s, color="Site1", title="Microcerotermes serratus Bray NMDS ps.tr")+stat_ellipse(type="norm", linetype=1)
#M_sp_G
ord.nmds.bray_micro2 <- ordinate(MspG1, method="NMDS", distance="bray")
phyloseq::plot_ordination(MspG1, ord.nmds.bray_micro2, color="Site1", title="Microcerotermes sp. G Bray NMDS ps.tr")


#Amitermes
ord.nmds.bray_ami <- ordinate(a_laurensis, method="NMDS", distance="bray")
a_laurensis_plot<-phyloseq::plot_ordination(a_laurensis, ord.nmds.bray_ami, color="Site1", title="Amitermes Bray NMDS genus")+ stat_ellipse(type="norm", linetype=1) +  theme_bw()#,label="X.NAME") #shape="Sequencing_group",
# with transformed data
ord.nmds.bray_ami1 <- ordinate(a_laurensis1, method="NMDS", distance="bray")
phyloseq::plot_ordination(a_laurensis1, ord.nmds.bray_ami1, shape="Site1",color="Sequencing_group", title="Amitermes Bray NMDS transformed")+stat_ellipse(type="norm", linetype=1)
plot_bar_plot(a_laurensis,.,Collection.site,.005)

#Nasutitermes
ord.nmds.bray_nasu <- ordinate(n_magnus, method="NMDS", distance="bray")
n_magnus_plot<-phyloseq::plot_ordination(n_magnus, ord.nmds.bray_nasu, color="Site1", title="Nasutitermes Bray NMDS genus")+ stat_ellipse(type="norm", linetype=1) +  theme_bw()#,label="X.NAME")shape="Sequencing_group",
ord.nmds.bray_nasu1 <- ordinate(n_magnus1, method="NMDS", distance="bray")
phyloseq::plot_ordination(n_magnus1, ord.nmds.bray_nasu1, color="Site1",shape="Sequencing_group", title="Nasutitermes Bray NMDS transformed")
plot_bar_plot(n_magnus,.,Site1,.005)

#Coptotermes
ord.nmds.bray_copto <- ordinate(c_acinaciformis, method="NMDS", distance="bray")
c_acinaciformis_plot<-phyloseq::plot_ordination(c_acinaciformis, ord.nmds.bray_copto, color="Site1", title="Coptotermes Bray NMDS genus")+ stat_ellipse(type="norm", linetype=1) +  theme_bw()#,label="X.NAME")
ord.nmds.bray_copto1 <- ordinate(c_acinaciformis1, method="NMDS", distance="bray")
phyloseq::plot_ordination(c_acinaciformis1, ord.nmds.bray_copto1, color="Site1", title="Coptotermes Bray NMDS transformed")
plot_bar_plot(c_acinaciformis,.,Collection.site,.005)

#zymo
ord.nmds.bray_zymo <- ordinate(zymo, method="NMDS", distance="bray")
phyloseq::plot_ordination(zymo, ord.nmds.bray_zymo, color="Sequencing_group", title="Bray NMDS",label="X.NAME")
plot_bar_plot(zymo1,.,Collection.site,0)
plot_bar(zymo1.tr, "Sample", "Abundance", 'Genus', labs(y="Relative Abundance", title = "Genus-level assignments"))+ facet_grid( .~ Sequencing_group, scales = "free", space = "free")

multiplot(a_laurensis_plot,m_serratus_plot,n_magnus_plot,c_acinaciformis_plot,cols=2)
```
Test for difference in betadiversity with adonis
```{r}
uniun<-phyloseq::distance(ps7, method="unifrac")
#uniun<-distance(Q_raref, method="unifrac")
#uniund<-as.matrix(uniun)
uniweigh<-phyloseq::distance(ps7, method="wunifrac")
#uniweigh<-distance(Q_raref, method="wunifrac")
#uniweighd<-as.matrix(uniweigh)
brayd<-phyloseq::distance(ps7, method="bray")
jaccd<-phyloseq::distance(ps7, method="jaccard")

p1 = phyloseq::plot_ordination(ps7, ordinate(ps7, method="PCoA", dist="wunifrac", weighted=TRUE), type = "samples", color = "t_sp") 
p1 + geom_point(size = 2) + ggtitle("PCoA Weighted UNIFRAC") #What is happening here? Why are coptotermes and Microcerotermes so close together?
p1 = phyloseq::plot_ordination(ps7, ordinate(ps7, method="PCoA", dist="unifrac"), type = "samples", color = "t_sp") 
p1 + geom_point(size = 2) + ggtitle("PCoA Unweighted UNIFRAC") 
p1 = phyloseq::plot_ordination(ps7, ordinate(ps7, method="PCoA", dist="brayd"), type = "samples", color = "t_sp") 
p1 + geom_point(size = 2) + ggtitle("PCoA Bray-Curtis") 
p1 = phyloseq::plot_ordination(ps7, ordinate(ps7, method="PCoA", dist="jaccd"), type = "samples", color = "t_sp") 
p1 + geom_point(size = 2) + ggtitle("PCoA Jaccard") 

```

Does physical distance determine distance in any of these groups? From Jeff Powell
```{r}
#I think the way to do this is to calculate multiple distance matrices and then convert each to a one-dimensional vector that can be stored in dataframe. Here that would be:

# 1) community distance matrix using dist() or vegdist()
uniun<-phyloseq::distance(ps7, method="bray")
# 2) termite species distance matrix – change termite to a factor, then as.numeric() so that euclidean distance can be calculated with dist(). Then interpret the result as 0 = same species, >0 = different species
termite_fac<-as.vector(ps7@sam_data[,4]) #species
termite_fac<-as.factor(termite_fac$t_sp)
fac_num<-as.numeric(termite_fac)
specdist<-dist(fac_num)

# 2b do the same thing for site
term_site<-as.factor(as.vector(ps7@sam_data[,5]$Site))
site_fac<-as.numeric(term_site)
sitedist<-dist(site_fac)

# 3) sampling site distance matrix – calculate Euclidean distances of geographic coordinates
?geodist()

df_distance<-as.data.frame(ps7@sam_data[,c(12,13)])
#colnames(df_distance)<-c("name","lat","lon")
df_distance<-as.matrix(df_distance)

mound_meters<-geodist(df_distance)

#Put in a dataframe, using as.numeric to change to one dimesion and lower.tri to account for mirroring of the matrix and the diagonal:
dim(uniun)
dim(brayd) #These should both be 118x118
 
commdist=as.numeric(brayd[lower.tri(brayd)])
specdist=as.numeric(specdist[lower.tri(specdist)])
sampledist=as.numeric(mound_meters[lower.tri(mound_meters)])
sitedist=as.numeric(sitedist[lower.tri(sitedist)])

model_df <- data.frame(commdist, specdist, sitedist, sampledist)

 
#Then how the model gets set up takes some thought. But I think you’d compare distance decay within termite species and between species:

model_df$samespec <- ifelse(model_df$specdist == 0, 'yes', 'no')
model_df$samesite <- ifelse(model_df$sitedist == 0, 'yes', 'no')

m1 <- lm(commdist ~ samespec + sampledist + samesite, data=model_df)
summary(m1) # samespec, sampledist and samesite all negatively coorelated with community distance.

ggplot(model_df,aes(sampledist,commdist)) +geom_point(aes(color=samespec,shape=samesite))


# Hmmm. I'd actually like to do this with each individual species. Change this from Unifrac (phylogenetic) to jaccard (presence absence) to Bray (includes relative abundance)
distance_phyloseq <- function(a_laurensis1) {
  commdist<-as.matrix(phyloseq::distance(a_laurensis1, method="bray"))
  commdist=as.numeric(commdist[lower.tri(commdist)])
  
  df_distance<-as.matrix(a_laurensis1@sam_data[,c(12,13)])
  mound_meters<-geodist(df_distance)
  sampledist=as.numeric(mound_meters[lower.tri(mound_meters)])
  
  term_site<-as.factor(as.vector(a_laurensis1@sam_data[,5]$Site))
  site_fac<-as.numeric(term_site)
  sitedist<-as.matrix(dist(site_fac,upper=T))
  sitedist1=as.numeric(sitedist[lower.tri(sitedist)])
  samesite <- ifelse(sitedist1 == 0, 'yes', 'no')
  
  termite_fac<-as.vector(a_laurensis1@sam_data[,4]) #species
  termite_fac<-as.factor(termite_fac$t_sp)
  fac_num<-as.numeric(termite_fac)
  specdist<-as.matrix(dist(fac_num,upper=T))
  specdist=as.numeric(specdist[lower.tri(specdist)])
  samespec <- ifelse(specdist == 0, 'yes', 'no')
  
  model_df <- data.frame(commdist, sampledist,samesite,samespec)
  
  return(model_df)
}
# For reference:
#PW-SC: 35km, PW-MS:38km, PW-JCU:86km, PW-RR:69km
#SC-MS: 4km, SC-JCU: 53km, SC-RR:52km
#MS-JCU:52km, MS-RR: 48km, JCU-RR: 77km

#Everything together
all_model<-distance_phyloseq(ps7)
ggplot(all_model,aes(sampledist,commdist)) +geom_point(aes(shape=samespec,color=samesite))+ggtitle("All species, all sites")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/allsp_allsites.png")
m1 <- lm(commdist ~ sampledist + samespec + samesite, data=all_model)
summary(m1) # sample distance is not significant. same site and same species have a negative association for unifrac.
#jaccard: same site: sample dist, samespecyes and samesiteyes all negatively correlated with community distance
#bray; all significant

a_model<-distance_phyloseq(a_laurensis1)
a_bothsites<-ggplot(a_model,aes(sampledist,commdist)) + geom_point(aes(color=samesite))+ggtitle("Amitermes laurensis")+geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?") #+stat_summary(fun.data=mean_cl_normal) +
   #geom_smooth(method='lm',formula=commdist ~  sampledist *samesite, data=a_model)
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Amitermes_allsites.png")
m2 <- lm(commdist ~  sampledist *samesite, data=a_model)
summary(m2) #sample dist and samesiteyes both have a negative correlation with community distance. no significant interaction
m2b <- lm(commdist ~  sampledist +samesite, data=a_model)
summary(m2b) #no significance for either
m2c <- lm(commdist ~  sampledist, data=a_model)
summary(m2c) #not significant
m2d <- lm(commdist ~  samesite, data=a_model)
summary(m2d) # yes significant
anova(m2,m2b) #not significant
anova(m2,m2d) 

c_model<-distance_phyloseq(c_acinaciformis1)
c_bothsites<-ggplot(c_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("Coptotermes acinaciformis")+geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Coptotermes_allsites.png")
m3 <- lm(commdist ~  sampledist +samesite, data=c_model)
summary(m3) # no significance or interaction
m3b<-lm(commdist ~  samesite, data=c_model)
summary(m3b) #yes significant
m3c<-lm(commdist ~  sampledist, data=c_model)
summary(m3c) #yes significant
anova(m3,m3b,m3c)

m_model<-distance_phyloseq(m_serratus1)
ggplot(m_model,aes(sampledist,commdist,color=samespec)) +geom_point(aes(shape=samesite))+ggtitle("Microcerotermes serratus")+geom_smooth(method="lm")+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same species?",shape="Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_allsites.png")
m4 <- lm(commdist ~  sampledist+samesite, data=m_model)
summary(m4) #sample dist and samesiteyes is negatively associated unifrac and bray and jaccard

ms_model<-distance_phyloseq(Mserratus)
m1_bothsites<-ggplot(ms_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("Microcerotermes serratus")+geom_smooth(method="lm")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_serratus_allsites.png")
m4s <- lm(commdist ~  sampledist+samesite, data=ms_model)
summary(m4s) #sample dist and samesiteyes is negatively associated unifrac and bray and jaccard
m4s <- lm(commdist ~  samesite, data=ms_model)
summary(m4s) # yes samesite is lower.

mG_model<-distance_phyloseq(MspG)
m2_bothsites<-ggplot(mG_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("Microcerotermes sp G")+geom_smooth(method="lm")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?")
m2_bothsites
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_spG_allsites.png")
m4 <- lm(commdist ~  sampledist+samesite, data=m_model)
summary(m4) #sample dist and samesiteyes is negatively associated unifrac and bray and jaccard

n_model<-distance_phyloseq(n_magnus1)
n_bothsites<-ggplot(n_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("Nasutitermes magnus")+geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Nasutitermes_allsites.png")
n_bothsites
m5 <- lm(commdist ~  sampledist+samesite, data=n_model)
summary(m5) #sampledist, samesiteyes have negative association, positive interaction for unifrac and bray.

multiplot(a_bothsites,m1_bothsites,n_bothsites,m2_bothsites,c_bothsites,cols=3)
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/multiplot_allsites.png")

# should we make a thing that says same site?
# All species, just two sites
pw_sc<-subset_samples(ps7,Site %in% c("SC","PW"))
pw_sc<-prune_taxa(taxa_sums(pw_sc)>0, pw_sc)
pw_sc_model<-distance_phyloseq(pw_sc)
ggplot(pw_sc_model,aes(sampledist,commdist,color=samespec)) +geom_point(aes(color=samespec,shape=samesite))+ggtitle("All species, savanna sites")+geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis community distance") + xlab("Geographical distance (m)")+ labs(color = "Same species?",shape="Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/allsp_Savanna.png")
m6 <- lm(commdist ~  samespec+sampledist+samesite, data=pw_sc_model)
summary(m6) #all significant unifrac. significant interaction between sample dist and samesite yes lollll
confint(m6)

# amitermes each site
a_pw<-subset_samples(a_laurensis1,Site=="PW")
a_pw_model<-distance_phyloseq(a_pw)
apw_plot<-ggplot(a_pw_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav1 Amitermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")+ labs(color = "Same site?")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Amitermes_PW.png")
m7 <- lm(commdist ~  sampledist, data=a_pw_model)
summary(m7) # not significant
a_sc<-subset_samples(a_laurensis1,Site=="SC")
a_sc_model<-distance_phyloseq(a_sc)
asc_plot<-ggplot(a_sc_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav2 Amitermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Amitermes_SC.png")
m8 <- lm(commdist ~  sampledist, data=a_sc_model)
summary(m8) #not significant

# nasutitermes each site
n_pw<-subset_samples(n_magnus1,Site=="PW")
n_pw_model<-distance_phyloseq(n_pw)
npw_plot<-ggplot(n_pw_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav1 Nasutitermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Nasutitermes_PW.png")
m9 <- lm(commdist ~  sampledist, data=n_pw_model)
summary(m9) # yes significant unifrac and jaccard and bray
n_sc<-subset_samples(n_magnus1,Site=="SC")
n_sc_model<-distance_phyloseq(n_sc)
nsc_plot<-ggplot(n_sc_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav2 Nasutitermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Nasutitermes_SC.png")
m10 <- lm(commdist ~  sampledist, data=n_sc_model)
summary(m10) # not significant
# coptotermes each site
c_pw<-subset_samples(c_acinaciformis1,Site=="PW")
c_pw_model<-distance_phyloseq(c_pw)
cpw_plot<-ggplot(c_pw_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav1 Coptotermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Coptotermes_PW.png")
m11 <- lm(commdist ~  sampledist, data=c_pw_model)
summary(m11) # not significant
c_sc<-subset_samples(c_acinaciformis1,Site=="SC")
c_sc_model<-distance_phyloseq(c_sc)
csc_plot<-ggplot(c_sc_model,aes(sampledist,commdist)) +geom_point()+ggtitle("Sav2 Coptotermes")+ geom_smooth(method='lm')+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Coptotermes_SC.png")
m12 <- lm(commdist ~  sampledist, data=c_sc_model)
summary(m12) #not significant
multiplot(apw_plot,npw_plot,cpw_plot,asc_plot,nsc_plot,csc_plot,cols=2)

# microcerotermes each site
m_pw<-subset_samples(m_serratus1,Site=="PW")
m_pw_model<-distance_phyloseq(m_pw)
ggplot(m_pw_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("PW Microcerotermes")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_PW.png")
m13 <- lm(commdist ~  sampledist, data=m_pw_model)
summary(m13) # not significant
m_sc<-subset_samples(m_serratus1,Site=="SC")
m_sc_model<-distance_phyloseq(m_sc)
ggplot(m_sc_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("SC Microcerotermes")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_SC.png")
m14 <- lm(commdist ~  sampledist, data=m_sc_model)
summary(m14) #not significant
m_ms<-subset_samples(m_serratus1,Site=="MS")
m_ms_model<-distance_phyloseq(m_ms)
ggplot(m_ms_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("MS Microcerotermes")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_MS.png")
m15 <- lm(commdist ~  sampledist, data=m_ms_model)
summary(m15) #not significant
m_jc<-subset_samples(m_serratus1,Site=="JC")
m_jc_model<-distance_phyloseq(m_jc)
ggplot(m_jc_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("JC Microcerotermes")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_JC.png")
m16 <- lm(commdist ~  sampledist, data=m_jc_model)
summary(m16) # not significant
m_rr<-subset_samples(m_serratus1,Site=="RR")
m_rr_model<-distance_phyloseq(m_rr)
ggplot(m_rr_model,aes(sampledist,commdist)) +geom_point(aes(color=samesite))+ggtitle("RR Microcerotermes")+theme_classic()+ylab("Bray-Curtis distance") + xlab("Geographical distance (m)")
ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/distances_bray/Microcerotermes_RR.png")
m17 <- lm(commdist ~  sampledist, data=m_rr_model)
summary(m17) #not significant with unifrac, but significant with jaccard and bray
```
Heat Map
```{r}
Phylum_Genus = apply(tax_table(ps.bygenus.tr.f)[, myranks], 1, paste, sep="", collapse=": ")
# Add concatenated labels as a new rank after strain
tax_table(ps.bygenus.tr.f) <- cbind(tax_table(ps.bygenus.tr.f), catglab=Phylum_Genus)
plot_heatmap(ps.bygenus.tr.f, sample.label="t_sp1", taxa.label = "catglab")
```
Scatter plot for each site
```{r}
ggplot(metadata,aes(Lat, Lon,color=t_sp))+geom_point()
# PW
ggplot(metadata[which[metadata$Site=="PW",],],aes(Lat, Lon,color=t_sp))+geom_point()
```


Compare models
```{r}
# How you compare models in adonis?
# Use AIC=2k+n[Ln(2(pi)*RSS/n)+1] and pick the model with lowest AIC score
# k=parameters, n=samples, RSS=SumsOfSqs

t1<-adonis(brayd~diversityRAREF_Q1$Site, strata=diversityRAREF_Q1$t_sp,perm=10000); t1 #r2=.18, p<0.05
t2<-adonis(brayd~diversityRAREF_Q1$Cluster, strata=diversityRAREF_Q1$t_sp,perm=10000); t2 #same, probably bc of unclustered ones from SC
t3<-adonis(brayd~diversityRAREF_Q1$Habitat, strata=diversityRAREF_Q1$t_sp,perm=10000); t3 #R2=.13, p<.05
t4<-adonis(brayd~diversityRAREF_Q1$Feeding_Group, strata=diversityRAREF_Q1$t_sp,perm=10000); t4 #.09 p=1 not significant
t5<-adonis(brayd~diversityRAREF_Q1$t_fam, strata=diversityRAREF_Q1$t_sp,perm=10000); t5 # R2=.06, p=1 not sig
t6<-adonis(brayd~diversityRAREF_Q1$Subfamily, strata=diversityRAREF_Q1$t_sp,perm=10000); t6 #R2=.16, p=1


t7<-adonis(uniweigh~diversityRAREF_Q1$t_sp1+diversityRAREF_Q1$Feeding_Group+diversityRAREF_Q1$Subfamily, strata=diversityRAREF_Q1$Mound_ID,perm=10000); t7

t1<-adonis(brayd~diversityRAREF_Q1$category+diversityRAREF_Q1$Feeding_Route1, strata=diversityRAREF_Q1$patient1,perm=10000); t1
t1<-adonis(jaccd~diversityRAREF_Q1$category+diversityRAREF_Q1$Feeding_Route1, strata=diversityRAREF_Q1$patient1,perm=10000); t1
```

## 4) Compare beta diversity with ordination. (ex. NMDS)
```{r}
# Transform data to proportions as appropriate for Bray-Curtis distances
ps.prop_all <- transform_sample_counts(ps8, function(otu) otu/sum(otu))
ord.nmds.bray_all <- ordinate(ps.prop_all, method="NMDS", distance="bray")

phyloseq::plot_ordination(ps.prop_all, ord.nmds.bray_all, color="t_sp",shape="Site1", title="Bray NMDS")+  theme_bw()#+ stat_ellipse(type="norm", linetype=1)  #

stressplot(ord.nmds.bray_all)

```


Use Deseq to see which taxa are different between Feeding group
csv with B are made from ps7. without are from ps.bygenus. NOte that those listed as grass could be grass or litter feeding.
```{r}
diagdds=phyloseq_to_deseq2(ps.bygenus,~Feeding_Group)
# calculate geometric means prior to estimate size factors
gm_mean = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans = apply(counts(diagdds), 1, gm_mean)
# Estimate size factors
diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
diagdds = DESeq(diagdds, test="Wald", fitType="local")
alpha = 0.05
res = results(diagdds, contrast = c("Feeding_Group", "Grass", "Wood"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res = res[order(res$padj, na.last=NA), ]
res
sigtab = res[(res$padj < alpha), ]
sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(ps7)[rownames(sigtab), ], "matrix"))
View(sigtab)
write.csv(sigtab,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_feeding_group_B.csv")

nrow(res) #264
nrow(ps.bygenus@otu_table) #264
nrow(sigtab) #3352 total ASV, 114 genera
```
Which taxa are different between subfamilies
```{r}
diagdds2=phyloseq_to_deseq2(ps.bygenus,~Subfamily)
# calculate geometric means prior to estimate size factors
gm_mean2 = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans2 = apply(counts(diagdds2), 1, gm_mean2)
# Estimate size factors
diagdds2 = estimateSizeFactors(diagdds2, geoMeans = geoMeans2)
diagdds2 = DESeq(diagdds2, test="Wald", fitType="local")

res2 = results(diagdds2, contrast = c("Subfamily","Termitinae", "Nasutitermitinae"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res2 = res2[order(res2$padj, na.last=NA), ]
View(res2)
alpha = 0.05
#alpha = 1.1
sigtab2 = res2[(res2$padj < alpha), ]
sigtab2 = cbind(as(sigtab2, "data.frame"), as(tax_table(ps7)[rownames(sigtab2), ], "matrix"))
View(sigtab2)
write.csv(sigtab2,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_subfamily_B.csv")
nrow(sigtab2) # 2882 ASVs, 95 genera
```
What taxa are different between species
```{r}
diagdds3=phyloseq_to_deseq2(ps.bygenus,~t_sp1)
# calculate geometric means prior to estimate size factors
gm_mean3 = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans3 = apply(counts(diagdds3), 1, gm_mean3)
# Estimate size factors
diagdds3 = estimateSizeFactors(diagdds3, geoMeans = geoMeans3)
diagdds3 = DESeq(diagdds3, test="Wald", fitType="local")

res3 = results(diagdds3, contrast = c("t_sp1","Amitermes","Nasutitermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res3 = res3[order(res3$padj, na.last=NA), ]
View(res3)
alpha = 0.05
#alpha = 1.1
sigtab3 = res3[(res3$padj < alpha), ]
sigtab3 = cbind(as(sigtab3, "data.frame"), as(tax_table(ps7)[rownames(sigtab3), ], "matrix"))
View(sigtab3)
write.csv(sigtab3,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Nasu_B.csv")
nrow(sigtab3) #3029 ASVs, 110 genera

res4 = results(diagdds3, contrast = c("t_sp1","Amitermes","Microcerotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res4 = res4[order(res4$padj, na.last=NA), ]
View(res4)
alpha = 0.05
#alpha = 1.1
sigtab4 = res4[(res4$padj < alpha), ]
sigtab4 = cbind(as(sigtab4, "data.frame"), as(tax_table(ps7)[rownames(sigtab4), ], "matrix"))
#View(sigtab4)
write.csv(sigtab4,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Micro_B.csv")
nrow(sigtab4) #126 genera, 2946 ASVs

res5 = results(diagdds3, contrast = c("t_sp1","Nasutitermes","Microcerotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res5 = res5[order(res5$padj, na.last=NA), ]
#View(res5)
alpha = 0.05
#alpha = 1.1
sigtab5 = res5[(res5$padj < alpha), ]
sigtab5 = cbind(as(sigtab5, "data.frame"), as(tax_table(ps7)[rownames(sigtab5), ], "matrix"))
View(sigtab5)
write.csv(sigtab5,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Nasu_Micro_B.csv")
nrow(sigtab5) #3350 ASVs, 106 genera

#Amitermes vs coptotermes
res6 = results(diagdds3, contrast = c("t_sp1","Amitermes","Coptotermes"),alpha=alpha); res6 = res6[order(res6$padj, na.last=NA), ];sigtab6 = res6[(res6$padj < alpha), ];sigtab6 = cbind(as(sigtab6, "data.frame"), as(tax_table(ps7)[rownames(sigtab6), ], "matrix"))
write.csv(sigtab6,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Copto_B.csv")
nrow(sigtab6) #135

#Nasutitermes vs coptotermes
res7 = results(diagdds3, contrast = c("t_sp1","Nasutitermes","Coptotermes"),alpha=alpha); res7 = res7[order(res7$padj, na.last=NA), ];sigtab7 = res7[(res7$padj < alpha), ];sigtab7 = cbind(as(sigtab7, "data.frame"), as(tax_table(ps7)[rownames(sigtab7), ], "matrix"))
write.csv(sigtab7,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Nasu_Copto_B.csv")
nrow(sigtab7) #116

#Microcerotermes vs coptotermes
res8 = results(diagdds3, contrast = c("t_sp1","Microcerotermes","Coptotermes"),alpha=alpha); res8 = res8[order(res8$padj, na.last=NA), ];sigtab8 = res8[(res8$padj < alpha), ];sigtab8 = cbind(as(sigtab8, "data.frame"), as(tax_table(ps7)[rownames(sigtab8), ], "matrix"))
write.csv(sigtab8,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Micro_Copto_B.csv")
nrow(sigtab8) #120

nrow(sigtab) #3352
nrow(sigtab2) #2882
nrow(sigtab3) #3029
nrow(sigtab4) #2946
nrow(sigtab5) #3350

otu_w_taxa<-cbind(ps7@otu_table,as(tax_table(ps7)[rownames(ps7@otu_table),],"matrix"))
write.csv(otu_w_taxa,"~/Box/Clement_TermiteGut_16S_0108/output/deseq/otu_w_taxa_ps7.csv")

genus_w_taxa<-cbind(ps.bygenus@otu_table,as(tax_table(ps.bygenus)[rownames(ps.bygenus@otu_table),],"matrix"))
write.csv(genus_w_taxa,"~/Box/Clement_TermiteGut_16S_0108/output/deseq/otu_w_taxa_genus.csv")

```


Plot differences
```{r}
scale_fill_discrete <- function(palname = "Set1", ...) {
  scale_fill_brewer(palette = palname, ...)
}
# Phylum order
x = tapply(sigtab3$log2FoldChange, sigtab3$Phylum, function(x) max(x))
x = sort(x, TRUE)
sigtab3$Phylum = factor(as.character(sigtab3$Phylum), levels=names(x))

# Species order
x = tapply(sigtab3$log2FoldChange, sigtab3$Genus, function(x) max(x))
x = sort(x, TRUE)
sigtab3$Genus = factor(as.character(sigtab3$Genus), levels=names(x))

ggplot(sigtab3, aes(x=Genus, y=log2FoldChange, color=Phylum)) + geom_point(size=6) + 
  theme(axis.text.x = element_text(angle = -90, hjust = 0, vjust=0.5))

```

Site differences for microcerotermes
```{r}

```

DESEQ at the phyla level
```{r}
ps.byphylum<-tax_glom(ps7,taxrank='Phylum')
nrow(ps.byphylum@otu_table) #30
diagdds4=phyloseq_to_deseq2(ps.byphylum,~t_sp1)
# calculate geometric means prior to estimate size factors
gm_mean4 = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans4 = apply(counts(diagdds4), 1, gm_mean4)
# Estimate size factors
diagdds4 = estimateSizeFactors(diagdds4, geoMeans = geoMeans4)
diagdds4 = DESeq(diagdds4, test="Wald", fitType="local")

res_p1 = results(diagdds4, contrast = c("t_sp1","Amitermes","Microcerotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p1 = res_p1[order(res_p1$padj, na.last=NA), ]
sigtab_p1 = res_p1[(res_p1$padj < alpha), ]
sigtab_p1 = cbind(as(sigtab_p1, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p1), ], "matrix"))
nrow(sigtab_p1) #22
sigtab_p1
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Micro_p1.csv")

res_p2 = results(diagdds4, contrast = c("t_sp1","Amitermes","Nasutitermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p2 = res_p2[order(res_p2$padj, na.last=NA), ]
sigtab_p2 = res_p2[(res_p2$padj < alpha), ]
sigtab_p2 = cbind(as(sigtab_p2, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p2), ], "matrix"))
nrow(sigtab_p2) #20
sigtab_p1
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Nasu_p1.csv")

res_p3 = results(diagdds4, contrast = c("t_sp1","Amitermes","Coptotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p3 = res_p3[order(res_p3$padj, na.last=NA), ]
sigtab_p3 = res_p3[(res_p3$padj < alpha), ]
sigtab_p3 = cbind(as(sigtab_p3, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p3), ], "matrix"))
nrow(sigtab_p1) #22
sigtab_p3
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Ami_Copto_p1.csv")

res_p4 = results(diagdds4, contrast = c("t_sp1","Microcerotermes","Nasutitermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p4 = res_p4[order(res_p4$padj, na.last=NA), ]
sigtab_p4 = res_p4[(res_p4$padj < alpha), ]
sigtab_p4 = cbind(as(sigtab_p4, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p4), ], "matrix"))
nrow(sigtab_p4) #16
sigtab_p4
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Micro_Nasu_p1.csv")

res_p5 = results(diagdds4, contrast = c("t_sp1","Microcerotermes","Coptotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p5 = res_p5[order(res_p5$padj, na.last=NA), ]
sigtab_p5 = res_p5[(res_p5$padj < alpha), ]
sigtab_p5 = cbind(as(sigtab_p5, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p5), ], "matrix"))
nrow(sigtab_p5) #20
sigtab_p5
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Micro_Copto_p1.csv")

res_p6 = results(diagdds4, contrast = c("t_sp1","Nasutitermes","Coptotermes"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_p6 = res_p6[order(res_p6$padj, na.last=NA), ]
sigtab_p6 = res_p6[(res_p6$padj < alpha), ]
sigtab_p6 = cbind(as(sigtab_p6, "data.frame"), as(tax_table(ps7)[rownames(sigtab_p6), ], "matrix"))
nrow(sigtab_p6) #20
sigtab_p6
write.csv(sigtab_p1,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_Nasu_Copto_p1.csv")

# Feeding group and subfamily now
diagdds5=phyloseq_to_deseq2(ps.byphylum,~Feeding_Group)
# calculate geometric means prior to estimate size factors
gm_mean5 = function(x, na.rm=TRUE){
  exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
}
geoMeans5 = apply(counts(diagdds5), 1, gm_mean5)
# Estimate size factors
diagdds5 = estimateSizeFactors(diagdds5, geoMeans = geoMeans5)
diagdds5 = DESeq(diagdds5, test="Wald", fitType="local")
res_fg = results(diagdds5, contrast = c("Feeding_Group", "Grass", "Wood"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_fg = res_fg[order(res_fg$padj, na.last=NA), ]
sigtab_fg = res_fg[(res_fg$padj < alpha), ]
sigtab_fg = cbind(as(sigtab_fg, "data.frame"), as(tax_table(ps7)[rownames(sigtab_fg), ], "matrix"))
write.csv(sigtab_fg,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_feeding_group_phylm.csv")
nrow(sigtab_fg) #12

#Now subfamily
diagdds6=phyloseq_to_deseq2(ps.byphylum,~Subfamily)
# calculate geometric means prior to estimate size factors
geoMeans6 = apply(counts(diagdds6), 1, gm_mean)
# Estimate size factors
diagdds6 = estimateSizeFactors(diagdds6, geoMeans = geoMeans6)
diagdds6 = DESeq(diagdds6, test="Wald", fitType="local")
res_sf = results(diagdds6, contrast = c("Subfamily", "Termitinae", "Nasutitermitinae"),alpha=alpha)
#res = results(diagdds, cooksCutoff = FALSE)
res_sf = res_sf[order(res_sf$padj, na.last=NA), ]
sigtab_sf = res_sf[(res_sf$padj < alpha), ]
sigtab_sf = cbind(as(sigtab_sf, "data.frame"), as(tax_table(ps7)[rownames(sigtab_sf), ], "matrix"))
write.csv(sigtab_sf,file="~/Box/Clement_TermiteGut_16S_0108/output/deseq/DESeqTaxa_subfamily_phylm.csv")
nrow(sigtab_sf) #14
```

Export biom files for picrust2 from https://www.youtube.com/watch?v=xZ7yc-GKcSk
```{r}
otu<-as(otu_table(ps8),"matrix")
otu_biom<-make_biom(data=otu)
write_biom(otu_biom,"~/Box/termite_microbiome/picrust2/otu.biom")
#also copy ASV_seqs_test_conc.fa, but just the rows that are still in ps8
fna_file<-readDNAStringSet("~/Box/termite_microbiome/Dada2/Jan2022/ASV_seqs_test_conc.fa")
fna_filt<-fna_file[rownames(ps8@otu_table),]
writeXStringSet(fna_filt,"~/Box/termite_microbiome/picrust2/fna_filt.fna",format="fasta")
# After installing picrust2 on a computer that has at least 16 Ram, run
# picrust2_pipeline.py -s fna_filt.fna -i otu.biom -o picrust2_out_pipeline -p 4
```

Identify core microbiome. How many ASVs are shared between all termites? How many are endemic? How many are similar between wood feeders? Upset plots? Occupancy abundance plots
```{r}
#https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#using_phyloseq
table(tax_table(ps8)[, "Phylum"], exclude = NULL)
abund_prev_plot <- function(ps8) {
  ps <- subset_taxa(ps8, !is.na(Phylum) & !Phylum %in% c("", "uncharacterized"))
  # Compute prevalence of each feature, store as data.frame
  prevdf = apply(X = otu_table(ps),
                 MARGIN = ifelse(taxa_are_rows(ps), yes = 1, no = 2),
                 FUN = function(x){sum(x > 0)})
  # Add taxonomy and total read counts to this data.frame
  prevdf = data.frame(Prevalence = prevdf,
                      TotalAbundance = taxa_sums(ps))#,
                      #tax_table(ps))
  prevdf$Phylum<-as.vector(tax_table(ps)[,2])
  prevdf$Family<-tax_table(ps)[,5]
  prevdf$Genus<-tax_table(ps)[,6]
  # Are there taxa that are comprised of mostly low-prevalence features?
  sum_reads<-sum(prevdf$TotalAbundance)
 prev_rel<- plyr::ddply(prevdf, "Phylum", function(df1){cbind(mean(df1$Prevalence),sum(df1$Prevalence),sum(df1$TotalAbundance),sum(df1$TotalAbundance)/sum_reads)}) # add total abundance, relative abundance here.
 #sort by relative abundance column
 colnames(prev_rel)<-c("Phylum","Avg prevalence","Total prevalence","Total reads","Rel_abundance")
 prev_rel<-prev_rel[order(-prev_rel$Rel_abundance),]
 prev_rel$Rel_abundance<-percent(prev_rel$Rel_abundance,accuracy = .1)
  # Define phyla to filter --ones that are fewer than 100 total prevalence
  #filterPhyla = c("Bdellovibrionota", "Campilobacterota","Chloroflexi","Deferribacterota","MBNT15","Myxococcota","Patescibacteria","WPS-2")
  # Filter entries with unidentified Phylum.
  #ps_prev = subset_taxa(ps, !Phylum %in% filterPhyla)
  #ps_prev
  #prevdf1 = subset(prevdf, Phylum %in% get_taxa_unique(ps_prev, "Phylum"))
  prevdf1 = subset(prevdf, Phylum %in% get_taxa_unique(ps, "Phylum"))
  ggplot(prevdf1, aes(TotalAbundance, Prevalence / nsamples(ps),color=Phylum)) +
    # Include a guess for parameter
    geom_hline(yintercept = 0.05, alpha = 0.5, linetype = 2) +  geom_point(size = 2, alpha = 0.7) +
    scale_x_log10() +  xlab("Total Abundance") + ylab("Prevalence [Frac. Samples]") +
    facet_wrap(~Phylum) + theme(legend.position="none")+ggtitle(ps@sam_data$t_sp1)
  #return(prev_rel)
  
}
abund_prev_plot(ps8)
abund_prev_plot(m_serratus1)
abund_prev_plot(m_serratus) #clustered by genus
abund_prev(a_laurensis1)
abund_prev(a_laurensis)
abund_prev(n_magnus1)
abund_prev(n_magnus)
nas_prevrel<-prevdf1
abund_prev(c_acinaciformis1)
abund_prev(c_acinaciformis)
c_prevdf<-prevdf1

```
Shared taxa
```{r}
ps8 <- prune_taxa(taxa_sums(ps7)>0, ps7)
nrow(ps8@otu_table) 
count_shared<-function(m_serratus1){
  # Compute prevalence of each feature, store as data.frame
  ps<-prune_taxa(taxa_sums(m_serratus1)>0, m_serratus1)
  prevdf = apply(X = otu_table(ps),
                 MARGIN = ifelse(taxa_are_rows(ps), yes = 1, no = 2),
                 FUN = function(x){sum(x > 0)})
  # Add taxonomy and total read counts to this data.frame
  prevdf = data.frame(Prevalence = prevdf,
                      TotalAbundance = taxa_sums(ps))#,
                      #tax_table(ps))
  prevdf<-cbind(as(prevdf,"data.frame"),as(tax_table(ps),"matrix"))
  sum_reads<-sum(prevdf$TotalAbundance)
  n_samp<-ncol(ps@otu_table)
  pct_prevalence<-percent(prevdf$Prevalence/n_samp,accuracy = .01)
  pct_abundance<-percent(prevdf$TotalAbundance/sum_reads,accuracy=.01)
 prev_rel<- cbind(prevdf,pct_prevalence,pct_abundance)
 #prev_rel<-prev_rel[order(-prev_rel$TotalAbundance),]
  # function to calculate number of shared ASVs
sortedprev<-prevdf[order(-prevdf$Prevalence),]
 
  n1<-nrow(sortedprev[sortedprev$Prevalence==n_samp,]) # How many ASVs are in all the samples?
  n.75<-nrow(sortedprev[sortedprev$Prevalence>n_samp*.75,])
  n2<-nrow(sortedprev[sortedprev$Prevalence>n_samp/2,]) # How many ASVs are in 50% the samples?
  n.25<-nrow(sortedprev[sortedprev$Prevalence>n_samp*.25,])
  print(paste("Number of samples:",n_samp,", Total ASVs:", length(prevdf$Prevalence>0),", Shared across all:",n1,", Shared across >75%:",n.75,", Shared across >50%:",n2,", Shared across >25%:",n.25))
  #return(prev_rel)
  }
count_shared(ps8) #There are 19017 ASVs in the table. Of these, 0 are prevalent across all 118 samples, 0 are prevalent in >75% of samples, and  9 are prevalent across >50% of samples."
count_shared(m_serratus1) #There are 4448 ASVs in the table. Of these, 18 are prevalent across all 40 samples, 37 are prevalent in >75% of samples, and  171 are prevalent across >50% of samples."
count_shared(a_laurensis1) #There are 7652 ASVs in the table. Of these, 8 are prevalent across all 28 samples, 63 are prevalent in >75% of samples, and  191 are prevalent across >50% of samples."
count_shared(n_magnus1) #There are 4534 ASVs in the table. Of these, 32 are prevalent across all 29 samples, 122 are prevalent in >75% of samples, and  496 are prevalent across >50% of samples."
count_shared(c_acinaciformis1) #There are 4069 ASVs in the table. Of these, 23 are prevalent across all 22 samples, 121 are prevalent in >75% of samples, and  388 are prevalent across >50% of samples."
count_shared(Mserratus)
count_shared(MspG)
# Now Genus level
count_shared(ps.bygenus)
count_shared(a_laurensis)
count_shared(n_magnus)
count_shared(c_acinaciformis)
count_shared(m_serratus)
count_shared(subset_samples(ps.bygenus,t_sp=="M_sp_G"))
count_shared(subset_samples(ps.bygenus,t_sp=="M_serratus"))
ps.byfamily<-tax_glom(ps8,taxrank="Family")
ps.byorder<-tax_glom(ps8,taxrank="Order")
ps.byphylum<-tax_glom(ps8,taxrank="Phylum")
# family level
count_shared(ps.byfamily)
count_shared(subset_samples(ps.byfamily,t_sp=="A_laurensis"))
count_shared(subset_samples(ps.byfamily,t_sp=="N_magnus"))
count_shared(subset_samples(ps.byfamily,t_sp=="C_acinaciformis"))
count_shared(subset_samples(ps.byfamily,t_sp1=="Microcerotermes"))
count_shared(subset_samples(ps.byfamily,t_sp=="M_serratus"))
count_shared(subset_samples(ps.byfamily,t_sp=="M_sp_G"))
count_shared(subset_samples(ps.byfamily,t_sp!="C_acinaciformis"))
count_shared(ps.byorder)
count_shared(subset_samples(ps.byorder,t_sp=="A_laurensis"))
count_shared(subset_samples(ps.byorder,t_sp=="N_magnus"))
count_shared(subset_samples(ps.byorder,t_sp=="C_acinaciformis"))
count_shared(subset_samples(ps.byorder,t_sp1=="Microcerotermes"))
count_shared(subset_samples(ps.byorder,t_sp=="M_serratus"))
count_shared(subset_samples(ps.byorder,t_sp=="M_sp_G"))
count_shared(subset_samples(ps.byorder,t_sp!="C_acinaciformis"))
phy1<-count_shared(ps.byphylum)
phy2<-count_shared(subset_samples(ps.byphylum,t_sp=="A_laurensis"))
phy2sav1<-count_shared(subset_samples(ps.byphylum,t_sp=="A_laurensis"&Site2=="Sav1"))
phy3<-count_shared(subset_samples(ps.byphylum,t_sp=="N_magnus"))
phy4<-count_shared(subset_samples(ps.byphylum,t_sp=="C_acinaciformis"))
count_shared(subset_samples(ps.byphylum,t_sp1=="Microcerotermes"))
phy5<-count_shared(subset_samples(ps.byphylum,t_sp=="M_serratus"))
phy6<-count_shared(subset_samples(ps.byphylum,t_sp=="M_sp_G"))
count_shared(subset_samples(ps.byphylum,t_sp!="C_acinaciformis")) #all higher termites

# Shared taxa between multiple groups

  shared_compare <- function(ps8,name1, name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum) {
    count_shared(subset_samples(ps8,t_sp1==name1|t_sp1==name2))
    count_shared(subset_samples(ps.bygenus,t_sp1==name1|t_sp1==name2))
    count_shared(subset_samples(ps.byfamily,t_sp1==name1|t_sp1==name2))
    count_shared(subset_samples(ps.byorder,t_sp1==name1|t_sp1==name2))
    count_shared(subset_samples(ps.byphylum,t_sp1==name1|t_sp1==name2))
  }
name1<-"Microcerotermes";name2<-"Amitermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
name1<-"Nasutitermes";name2<-"Amitermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
name1<-"Microcerotermes";name2<-"Nasutitermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
name1<-"Coptotermes";name2<-"Amitermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
name1<-"Coptotermes";name2<-"Nasutitermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
name1<-"Coptotermes";name2<-"Microcerotermes"
shared_compare(ps8,name1,name2, ps.bygenus, ps.byfamily, ps.byorder, ps.byphylum)
# same but just for sav1 and just for sav2

# check 3 at a time
count_shared(subset_samples(ps.bygenus,t_sp1=="Amitermes"|t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Amitermes"|t_sp1=="Coptotermes"|t_sp1=="Microcerotermes"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Amitermes"|t_sp1=="Nasutitermes"|t_sp1=="Coptotermes"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Coptotermes"|t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes"))
count_shared(subset_samples(ps.byphylum,t_sp1=="Amitermes"|t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes"))
# Check at each site
count_shared(subset_samples(ps8,Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,Site2=="Sav1"))
count_shared(subset_samples(ps.byfamily,Site2=="Sav1"))
count_shared(subset_samples(ps.byorder,Site2=="Sav1"))
count_shared(subset_samples(ps.byphylum,Site2=="Sav1"))

count_shared(subset_samples(ps8,Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,Site2=="Sav2"))
count_shared(subset_samples(ps.byfamily,Site2=="Sav2"))
count_shared(subset_samples(ps.byorder,Site2=="Sav2"))
count_shared(subset_samples(ps.byphylum,Site2=="Sav2"))

#Each species at each site
count_shared_sites <- function(ps8) {
  count_shared(subset_samples(ps8,t_sp1=="Amitermes"&Site2=="Sav1"))
  count_shared(subset_samples(ps8,t_sp1=="Amitermes"&Site2=="Sav2"))
  count_shared(subset_samples(ps8,t_sp1=="Nasutitermes"&Site2=="Sav1"))
  count_shared(subset_samples(ps8,t_sp1=="Nasutitermes"&Site2=="Sav2"))
  count_shared(subset_samples(ps8,t_sp1=="Coptotermes"&Site2=="Sav1"))
  count_shared(subset_samples(ps8,t_sp1=="Coptotermes"&Site2=="Sav2"))
  count_shared(subset_samples(ps8,t_sp1=="Microcerotermes"&Site2=="Sav1"))
  count_shared(subset_samples(ps8,t_sp1=="Microcerotermes"&Site2=="Sav2"))
  count_shared(subset_samples(ps8,t_sp1=="Microcerotermes"&Site2=="Scl1"))
  count_shared(subset_samples(ps8,t_sp1=="Microcerotermes"&Site2=="RftJ"))
  count_shared(subset_samples(ps8,t_sp1=="Microcerotermes"&Site2=="RftR"))
}
#Genus level
count_shared(subset_samples(ps.bygenus,t_sp1=="Amitermes"&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Amitermes"&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Nasutitermes"&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Nasutitermes"&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Coptotermes"&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Coptotermes"&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Microcerotermes"&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Microcerotermes"&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Microcerotermes"&Site2=="Scl1"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Microcerotermes"&Site2=="RftJ"))
count_shared(subset_samples(ps.bygenus,t_sp1=="Microcerotermes"&Site2=="RftR"))
# Family level
count_shared_sites(ps.byfamily)
# Order level
count_shared_sites(ps.byorder)
# Phylum level
count_shared_sites(ps.byphylum)
# Genus level between species combos
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Nasutitermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Microcerotermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Coptotermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Coptotermes"|t_sp1=="Microcerotermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Coptotermes")&Site2=="Sav1"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes"|t_sp1=="Amitermes")&Site2=="Sav1"))
#Sav2
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Nasutitermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Microcerotermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Amitermes"|t_sp1=="Coptotermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Coptotermes"|t_sp1=="Microcerotermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Coptotermes")&Site2=="Sav2"))
count_shared(subset_samples(ps.bygenus,(t_sp1=="Nasutitermes"|t_sp1=="Microcerotermes"|t_sp1=="Amitermes")&Site2=="Sav2"))

```
Upset plot to show overlap in genera
```{r}
library(UpSetR) #https://www.r-graph-gallery.com/upset-plot.html
input75<-c(Amitermes = 65, Nasutitermes = 48, Coptotermes = 54, Microcerotermes = 40, "Amitermes&Nasutitermes"=44, "Amitermes&Microcerotermes"=34,"Amitermes&Coptotermes"=34,"Nasutitermes&Microcerotermes"=37,"Nasutitermes&Coptotermes"=30,"Microcerotermes&Coptotermes"=26,"Amitermes&Nasutitermes&Coptotermes"=30,"Amitermes&Nasutitermes&Microcerotermes"=37,"Nasutitermes&Microcerotermes&Coptotermes"=33,"Amitermes&Microcerotermes&Coptotermes"=30,"Amitermes&Microcerotermes&Coptotermes&Nasutitermes"=36)
input<-c(Amitermes = 31, Nasutitermes = 17, Coptotermes = 33, Microcerotermes = 12, "Amitermes&Nasutitermes"=8, "Amitermes&Microcerotermes"=7,"Amitermes&Coptotermes"=15,"Nasutitermes&Microcerotermes"=6,"Nasutitermes&Coptotermes"=8,"Microcerotermes&Coptotermes"=5,"Amitermes&Nasutitermes&Coptotermes"=7,"Amitermes&Nasutitermes&Microcerotermes"=4,"Nasutitermes&Microcerotermes&Coptotermes"=3,"Amitermes&Microcerotermes&Coptotermes"=5,"Amitermes&Microcerotermes&Coptotermes&Nasutitermes"=3)
upset(fromExpression(input), 
      nintersects = 40, 
      nsets = 6, 
      order.by = "freq", 
      decreasing = T, 
      mb.ratio = c(0.6, 0.4),
      number.angles = 0, 
      text.scale = 1.1, 
      point.size = 2.8, 
      line.size = 1
      )


dim(sets)
# Prep the data for a Venn diagram https://www.littlemissdata.com/blog/set-analysis

#Note that if you need to move around the labels so that they are not overlapping, you can use the new line breaks like the example below.
#v$labels <- c("TTC", "Walk", "Drive", "Cycle\n\n\n", "\nTaxi", "Community Ride", "Wheel Trans", "Friends")

#Merge samples to do this
ps8@sam_data$sp_habitat<-paste(ps8@sam_data$t_sp,ps8@sam_data$Site2)
merged_by_ss.phy = merge_samples(ps8, "sp_habitat")
ps.byphylum@sam_data$sp_habitat<-paste(ps.byphylum@sam_data$t_sp,ps.byphylum@sam_data$Site2)
merged_by_ss.phylum = merge_samples(ps.byphylum, "sp_habitat")
merged_by_ss.df <-as.data.frame(t(merged_by_ss.phy@otu_table))
dim(merged_by_ss.df)
merged_by_ss.df[merged_by_ss.df>0]<-1 #convert table to presence/absence
merged_by_ss.df$ASV<-rownames(merged_by_ss.df)
vennMerged <- merged_by_ss.df %>%
          gather(SpeciesSite, binary,1:11) %>% # take all binary mappings and convert to be a the set indicator
          filter(binary == 1) %>% # only include set matches
          select(ASV,SpeciesSite) %>% # only include ID and set category
          mutate(SpeciesSite = factor(SpeciesSite))
v <- venneuler(data.frame(vennMerged))

plot(v, main = "Shared ASV prevalence across species and sites", cex.main = 1)
upset_ASV<-upset(merged_by_ss.df,
  nsets = 11, number.angles = 30, point.size = 3.5, line.size = 2, order.by = "freq", 
      decreasing = T,
  mainbar.y.label = "Shared ASV prevalence across species and sites", sets.x.label = "Total ASVs"
)
ggsave("~/Box/termite_microbiome/Figures/UpsetPlot_ASV.png",height = 10,width=15)

# Now do the same but at the genus level
ps.bygenus@sam_data$sp_habitat<-paste(ps.bygenus@sam_data$t_sp,ps.bygenus@sam_data$Site2)
merged_by_ss.genus = merge_samples(ps.bygenus, "sp_habitat")
merged_by_ss.df.genus <-as.data.frame(t(merged_by_ss.genus@otu_table))
dim(merged_by_ss.df.genus) #264x11
table_merge<-cbind(tax_table(merged_by_ss.genus),t(otu_table(merged_by_ss.genus)))
write.csv(table_merge,file="~/Box/termite_microbiome/Dada2/Jan2022/table_tax_merged_genus.csv")

merged_by_ss.df.genus[merged_by_ss.df.genus>0]<-1 #convert table to presence/absence
merged_by_ss.df.genus$ASV<-rownames(merged_by_ss.df.genus)
vennMerged <- merged_by_ss.df.genus %>%
          gather(SpeciesSite, binary,1:11) %>% # take all binary mappings and convert to be a the set indicator
          filter(binary == 1) %>% # only include set matches
          select(ASV,SpeciesSite) %>% # only include ID and set category
          mutate(SpeciesSite = factor(SpeciesSite))
v <- venneuler(data.frame(vennMerged))

plot(v, main = "Shared genera prevalence across species and sites", cex.main = 1)
upset_genus<-upset(merged_by_ss.df.genus,
  nsets = 11, number.angles = 30, point.size = 3.5, line.size = 2, order.by = "freq", 
      decreasing = T,
  mainbar.y.label = "Shared genera prevalence across species and sites", sets.x.label = "Total genera"
)
ggsave("~/Box/termite_microbiome/Figures/UpsetPlot_genus.png",height = 10,width=15)
```

How many species in each site?
```{r}
Sav1<-ps8@sam_data[ps8@sam_data$Site2=="Sav1",]
table(Sav1$t_sp)
sav1<-metadata[metadata$Site=="PW",]
table(sav1$t_sp)
Sav2<-ps8@sam_data[ps8@sam_data$Site2=="Sav2",]
table(Sav2$t_sp)
sav2<-metadata[metadata$Site=="SC",]
table(sav2$t_sp)
table(ps8@sam_data$Site2[ps8@sam_data$t_sp=="M_sp_G"])
```

Genus level DESEQ between species at different sites
```{r}

deseq_seqtab <- function(a_laurensis, Site2,fsite1,fsite2) {
  diagdds=phyloseq_to_deseq2(a_laurensis,~Site2)
  # calculate geometric means prior to estimate size factors
  geoMeans = apply(counts(diagdds), 1, gm_mean)
  # Estimate size factors
  diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
  diagdds = DESeq(diagdds, test="Wald", fitType="local")
  alpha=.05
  res = results(diagdds, contrast = c("Site2",fsite1,fsite2),alpha=alpha)
  res = res[order(res$padj, na.last=NA), ]
  res
  sigtab = res[(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(ps7)[rownames(sigtab), ], "matrix")) #add taxonomy
  return(sigtab)
}
sigtab_sav_genus<-deseq_seqtab(ps.bygenus,Site2,"Sav1","Sav2")
nrow(sigtab_sav_genus)

sigtab_A<-deseq_seqtab(a_laurensis,Site2,"Sav1","Sav2")
nrow(sigtab_A)
write.csv(sigtab_A,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Ami_genus.csv")
sigtab_N<-deseq_seqtab(n_magnus,Site2,"Sav1","Sav2")
nrow(sigtab_N)
write.csv(sigtab_N,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Nas_genus.csv")
sigtab_C<-deseq_seqtab(c_acinaciformis,Site2,"Sav1","Sav2")
nrow(sigtab_C)
write.csv(sigtab_C,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Cop_genus.csv")
sigtab_M<-deseq_seqtab(m_serratus,Site2,"Sav1","Sav2")
nrow(sigtab_M)
write.csv(sigtab_M,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_sav2_genus.csv")
sigtab_M2<-deseq_seqtab(m_serratus,Site2,"Sav1","Scl1")
nrow(sigtab_M2)
write.csv(sigtab_M2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_scl1_genus.csv")
sigtab_M3<-deseq_seqtab(m_serratus,Site2,"Sav1","RftJ")
nrow(sigtab_M3)
sigtab_M4<-deseq_seqtab(m_serratus,Site2,"Sav1","RftR")
nrow(sigtab_M4)
sigtab_M5<-deseq_seqtab(m_serratus,Site2,"Scl1","RftJ")
nrow(sigtab_M5)
sigtab_M6<-deseq_seqtab(m_serratus,Site2,"Scl1","RftR")
nrow(sigtab_M6)
sigtab_M7<-deseq_seqtab(m_serratus,Site2,"RftJ","RftR")
nrow(sigtab_M7)
sigtab_M8<-deseq_seqtab(m_serratus,Site2,"Sav2","Scl1")
nrow(sigtab_M8)
sigtab_M9<-deseq_seqtab(m_serratus,Site2,"Sav2","RftJ")
nrow(sigtab_M9)
sigtab_M10<-deseq_seqtab(m_serratus,Site2,"Sav2","RftR")
nrow(sigtab_M10)

# ASV level differences
sigtab_sav<-deseq_seqtab(ps8,Site2,"Sav1","Sav2")
nrow(sigtab_sav)

sigtab_A2<-deseq_seqtab(a_laurensis2,Site2,"Sav1","Sav2")
nrow(sigtab_A2)
write.csv(sigtab_A2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Ami_ASV.csv")
sigtab_N2<-deseq_seqtab(n_magnus2,Site2,"Sav1","Sav2")
nrow(sigtab_N2)
write.csv(sigtab_N2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Nas_ASV.csv")
sigtab_C2<-deseq_seqtab(c_acinaciformis2,Site2,"Sav1","Sav2")
nrow(sigtab_C2)
write.csv(sigtab_C2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Cop_ASV.csv")
sigtab2_M<-deseq_seqtab(m_serratus2,Site2,"Sav1","Sav2")
nrow(sigtab2_M)
write.csv(sigtab2_M,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_sav2_ASV.csv")
sigtab2_M2<-deseq_seqtab(m_serratus2,Site2,"Sav1","Scl1")
nrow(sigtab2_M2)
write.csv(sigtab2_M2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_scl1_ASV.csv")
sigtab2_M3<-deseq_seqtab(m_serratus2,Site2,"Sav1","RftJ")
nrow(sigtab2_M3)
sigtab2_M4<-deseq_seqtab(m_serratus2,Site2,"Sav1","RftR")
nrow(sigtab2_M4)
sigtab2_M5<-deseq_seqtab(m_serratus2,Site2,"Scl1","RftJ")
nrow(sigtab2_M5)
sigtab2_M6<-deseq_seqtab(m_serratus2,Site2,"Scl1","RftR")
nrow(sigtab2_M6)
sigtab2_M7<-deseq_seqtab(m_serratus2,Site2,"RftJ","RftR")
nrow(sigtab2_M7)
sigtab2_M8<-deseq_seqtab(m_serratus2,Site2,"Sav2","Scl1")
nrow(sigtab2_M8)
sigtab2_M9<-deseq_seqtab(m_serratus2,Site2,"Sav2","Rft")
nrow(sigtab2_M9)
sigtab2_M10<-deseq_seqtab(m_serratus2,Site2,"Sav2","RftR")
nrow(sigtab2_M10)

#new function for feeding group
deseq_seqtab_feed <- function(a_laurensis, Feeding_Group) {
  diagdds=phyloseq_to_deseq2(a_laurensis,~Feeding_Group)
  # calculate geometric means prior to estimate size factors
  geoMeans = apply(counts(diagdds), 1, gm_mean)
  # Estimate size factors
  diagdds = estimateSizeFactors(diagdds, geoMeans = geoMeans)
  diagdds = DESeq(diagdds, test="Wald", fitType="local")
  alpha=.05
  res = results(diagdds, contrast = c("Feeding_Group","Wood","Grass"),alpha=alpha)
  res = res[order(res$padj, na.last=NA), ]
  res
  sigtab = res[(res$padj < alpha), ]
  sigtab = cbind(as(sigtab, "data.frame"), as(tax_table(ps7)[rownames(sigtab), ], "matrix")) #add taxonomy, take off this line if you are doing function
  return(sigtab)
}


sigtab_feed_genus<-deseq_seqtab_feed(ps.bygenus,Feeding_group)
nrow(sigtab_feed_genus)
head(sigtab_feed_genus[order(-sigtab_feed_genus$baseMean),c(1,2,5,6,8,12)],10)
sigtab_feed<-deseq_seqtab_feed(ps8,Feeding_group)
nrow(sigtab_feed)
sigtab_feed3<-deseq_seqtab_feed(ps9,Feeding_group)
nrow(sigtab_feed3)
head(sigtab_feed3[order(-sigtab_feed3$log2FoldChange),c(1,2,5,6)],10)
```

DESEQ with EC instead of genus
```{r}
  EC_table<-read.table("~/Box/termite_microbiome/picrust2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv",row.names=1, header=T)
nrow(EC_table) #423 pathways
tax9<-cbind(rownames(EC_table),rownames(EC_table))
colnames(tax9)<-c("Name","EC")
tax9<-as.data.frame(tax9)
row.names(tax9)<-tax9$Name
ps9<-phyloseq(sample_data(ps8@sam_data),tax_table(as.matrix(tax9)), otu_table(EC_table, taxa_are_rows = TRUE))

ord.nmds.bray_fun<- ordinate(ps9, method="NMDS", distance="bray")
phyloseq::plot_ordination(ps9, ord.nmds.bray_fun, color="t_sp",shape="Site2", title="Bray NMDS Pathway")+   theme_bw()+geom_polygon() + geom_point(size=5)#+stat_ellipse(type="norm", linetype=1) 

# Subset these
a_laurensis3<-subset_samples(ps9,t_sp1=="Amitermes") #with whole data
a_laurensis3<-prune_taxa(taxa_sums(a_laurensis3)>0, a_laurensis3)
nrow(a_laurensis3@otu_table)
n_magnus3<-subset_samples(ps9,t_sp1=="Nasutitermes") #with whole data
n_magnus3<-prune_taxa(taxa_sums(n_magnus3)>0, n_magnus3)
nrow(n_magnus3@otu_table)
c_acinaciformis3<-subset_samples(ps9,t_sp1=="Coptotermes") #with whole data
c_acinaciformis3<-prune_taxa(taxa_sums(c_acinaciformis3)>0, c_acinaciformis3)
nrow(c_acinaciformis3@otu_table)
m_serratus3<-subset_samples(ps9,t_sp1=="Microcerotermes") #with whole data
m_serratus3<-prune_taxa(taxa_sums(m_serratus3)>0, m_serratus3)
nrow(m_serratus3@otu_table)
mserratus3<-subset_samples(m_serratus3,t_sp=="M_serratus")
nrow(prune_taxa(taxa_sums(mserratus3)>0, mserratus3)@otu_table)
mspG3<-subset_samples(m_serratus3,t_sp=="M_sp_G")
nrow(prune_taxa(taxa_sums(mspG3)>0, mspG3)@otu_table)

savs<-subset_samples(ps8,Habitat=="Savanna")
nrow(prune_taxa(taxa_sums(savs)>0, savs)@otu_table)
savs_genus<-subset_samples(ps.bygenus,Habitat=="Savanna")
nrow(prune_taxa(taxa_sums(savs_genus)>0, savs_genus)@otu_table)
savs3<-subset_samples(ps9,Habitat=="Savanna")
nrow(prune_taxa(taxa_sums(savs3)>0, savs3)@otu_table)
sigtab_sav3<-deseq_seqtab(ps9,Site2,"Sav1","Sav2")
nrow(sigtab_sav3)

sigtab_A3<-deseq_seqtab(a_laurensis3,Site2,"Sav1","Sav2")
nrow(sigtab_A3)
sigtab_A3[,c(1,2,5,6)]
write.csv(sigtab_A3,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Ami_fun.csv")
sigtab_N3<-deseq_seqtab(n_magnus3,Site2,"Sav1","Sav2")
nrow(sigtab_N3)
head(sigtab_N3[order(-sigtab_N3$baseMean),c(1,2,5,6)],10)
write.csv(sigtab_N3,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Nas_fun.csv")
sigtab_C3<-deseq_seqtab(c_acinaciformis3,Site2,"Sav1","Sav2")
nrow(sigtab_C3)
write.csv(sigtab_C3,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Cop_fun.csv")
sigtab3_M<-deseq_seqtab(m_serratus3,Site2,"Sav1","Sav2")
nrow(sigtab3_M)
write.csv(sigtab3_M,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_sav2_fun.csv")
sigtab3_M2<-deseq_seqtab(m_serratus3,Site2,"Sav1","Scl1")
nrow(sigtab3_M2)
head(sigtab3_M2[order(-sigtab3_M2$baseMean),c(1,2,5,6)],10)
write.csv(sigtab3_M2,file="~/Box/termite_microbiome/DESEQ/DESeqTaxa_Mic_sav1_scl1_fun.csv")
sigtab3_M3<-deseq_seqtab(m_serratus3,Site2,"Sav1","RftJ")
nrow(sigtab3_M3)
sigtab3_M4<-deseq_seqtab(m_serratus3,Site2,"Sav1","RftR")
nrow(sigtab3_M4)
sigtab3_M5<-deseq_seqtab(m_serratus3,Site2,"Scl1","RftJ")
nrow(sigtab3_M5)
sigtab3_M6<-deseq_seqtab(m_serratus3,Site2,"Scl1","RftR")
nrow(sigtab3_M6)
sigtab3_M7<-deseq_seqtab(m_serratus3,Site2,"RftJ","RftR")
nrow(sigtab3_M7)
sigtab3_M8<-deseq_seqtab(m_serratus3,Site2,"Sav2","Scl1")
nrow(sigtab3_M8)
sigtab3_M9<-deseq_seqtab(m_serratus3,Site2,"Sav2","RftJ")
nrow(sigtab3_M9)
sigtab3_M10<-deseq_seqtab(m_serratus3,Site2,"Sav2","RftR")
nrow(sigtab3_M10)



#Upset plot
ps9@sam_data$sp_habitat<-paste(ps9@sam_data$t_sp,ps9@sam_data$Site2)
merged_by_ss.fun = merge_samples(ps9, "sp_habitat")
View(t(merged_by_ss.fun@otu_table)) # to see what most common are
merged_by_ss.df_fun <-as.data.frame(t(merged_by_ss.fun@otu_table))
dim(merged_by_ss.df_fun) #423x11
merged_by_ss.df_fun[merged_by_ss.df_fun>0]<-1 #convert table to presence/absence
merged_by_ss.df_fun$ASV<-rownames(merged_by_ss.df_fun)
vennMerged_fun <- merged_by_ss.df_fun %>%
          gather(SpeciesSite, binary,1:11) %>% # take all binary mappings and convert to be a the set indicator
          filter(binary == 1) %>% # only include set matches
          select(ASV,SpeciesSite) %>% # only include ID and set category
          mutate(SpeciesSite = factor(SpeciesSite))
v <- venneuler(data.frame(vennMerged_fun))

plot(v, main = "Shared pathways across species and sites", cex.main = 1)
upset_fun<-upset(merged_by_ss.df_fun,
  nsets = 11, number.angles = 30, point.size = 3.5, line.size = 2, order.by = "freq", 
      decreasing = T,
  mainbar.y.label = "Shared function prevalence across species and sites", sets.x.label = "Total EC functions"
)
upset_fun

```

Three barplots on top of each other https://jokergoo.github.io/ComplexHeatmap-reference/book/upset-plot.html
```{r}
#BiocManager::install("ComplexHeatmap")
m<-make_comb_mat(merged_by_ss.df,top_n_sets = 11)
m = m[comb_degree(m) > 0]
UpSet(m[comb_size(m) >= 30],set_order =NULL,comb_order = order(-comb_size(m[comb_size(m) >= 30])))

#by genus
m1<-make_comb_mat(merged_by_ss.df.genus,top_n_sets = 11)
m1 = m1[comb_degree(m1) > 0]
UpSet(m1[comb_size(m1) >= 2],set_order =NULL,comb_order = order(-comb_size(m1[comb_size(m1) >= 2])))

# By function
m2<-make_comb_mat(merged_by_ss.df_fun,top_n_sets = 11)
m2 = m2[comb_size(m2) >= 10]
ht<-UpSet(m2,set_order =NULL,comb_order = order(-comb_size(m2)),top_annotation = HeatmapAnnotation(
        "Intersection size" = anno_barplot(comb_size(m2), 
            ylim = c(0, max(comb_size(m2))*1.1),
            border = FALSE, 
            gp = gpar(fill = "black"), 
            height = unit(4, "cm")
        )))
ht = draw(ht)
od = column_order(ht)
decorate_annotation("Intersections", {
    grid.text(comb_size(m2)[od], x = seq_along(comb_size(m2)), y = unit(comb_size(m2)[od], "native") + unit(2, "pt"), 
        default.units = "native", just = c("left", "bottom"), 
        gp = gpar(fontsize = 6, col = "#404040"), rot = 45)
})

```

DESEQ by phylum
```{r}
nrow(ps.byphylum@otu_table)
sav_phylum<-subset_samples(ps.byphylum,Habitat=="Savanna")
nrow(prune_taxa(taxa_sums(sav_phylum)>0, sav_phylum)@otu_table)
a_laurensis_phylum<-subset_samples(ps.byphylum,t_sp1=="Amitermes")
nrow(prune_taxa(taxa_sums(a_laurensis_phylum)>0, a_laurensis_phylum)@otu_table)
n_magnus_phylum<-subset_samples(ps.byphylum,t_sp1=="Nasutitermes")
nrow(prune_taxa(taxa_sums(n_magnus_phylum)>0, n_magnus_phylum)@otu_table)
c_acinaciformis_phylum<-subset_samples(ps.byphylum,t_sp1=="Coptotermes")
nrow(prune_taxa(taxa_sums(c_acinaciformis_phylum)>0, c_acinaciformis_phylum)@otu_table)
m_serratus_phylum<-subset_samples(ps.byphylum,t_sp1=="Microcerotermes")
nrow(prune_taxa(taxa_sums(m_serratus_phylum)>0, m_serratus_phylum)@otu_table)
mserratus_phylum<-subset_samples(ps.byphylum,t_sp=="M_serratus")
nrow(prune_taxa(taxa_sums(mserratus_phylum)>0, mserratus_phylum)@otu_table)
mspG_phylum<-subset_samples(ps.byphylum,t_sp=="M_sp_G")
nrow(prune_taxa(taxa_sums(mspG_phylum)>0, mspG_phylum)@otu_table)

# now do DESEQ
sigtab_feed_phylum<-deseq_seqtab_feed(ps.byphylum,Feeding_group)
nrow(sigtab_feed_phylum)
sigtab_feed_phylum[,c("Phylum","log2FoldChange")]
sigtab_sav_phylum<-deseq_seqtab(ps.byphylum,Site2,"Sav1","Sav2")
nrow(sigtab_sav_phylum)
sigtab_sav_phylum[,c("Phylum","baseMean","log2FoldChange")]
sigtab_A_phylum<-deseq_seqtab(a_laurensis_phylum,Site2,"Sav1","Sav2")
nrow(sigtab_A_phylum)
sigtab_N_phylum<-deseq_seqtab(n_magnus_phylum,Site2,"Sav1","Sav2")
nrow(sigtab_N_phylum)
sigtab_C_phylum<-deseq_seqtab(c_acinaciformis_phylum,Site2,"Sav1","Sav2")
nrow(sigtab_C_phylum)
sigtab_M_phylum<-deseq_seqtab(mserratus_phylum,Site2,"Sav1","Sav2")
nrow(sigtab_M_phylum)
sigtab_M1_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav1","Scl1")
nrow(sigtab_M1_phylum)
sigtab_M2_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav1","RftJ")
nrow(sigtab_M2_phylum)
sigtab_M3_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav1","RftR")
nrow(sigtab_M3_phylum)
sigtab_M4_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Scl1","RftJ")
nrow(sigtab_M4_phylum)
sigtab_M5_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Scl1","RftR")
nrow(sigtab_M5_phylum)
sigtab_M6_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"RftR","RftJ")
nrow(sigtab_M6_phylum)
sigtab_M6_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"RftR","RftJ")
nrow(sigtab_M6_phylum)
sigtab_M7_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav2","Scl1")
nrow(sigtab_M7_phylum)
sigtab_M8_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav2","RftJ")
nrow(sigtab_M8_phylum)
sigtab_M9_phylum<-deseq_seqtab(m_serratus_phylum,Site2,"Sav2","RftR")
nrow(sigtab_M9_phylum)

```
Make plot of genus based on merged samples
```{r}
ps.any.tr <- transform_sample_counts(merged_by_ss.phylum, function (x) x / sum(x))
keep_percent<-.005
ps.any.f <- filter_taxa(ps.any.tr, function (x) mean(x) > keep_percent, TRUE) 
  patho.by_any_plot <- plot_bar(ps.any.f, "Sample", "Abundance", 'Phylum', labs(y="Relative Abundance", title = "Phylum-level assignments"))
  ps.by_phylum_plot <- patho.by_any_plot + 
    theme_classic() + 
    #facet_wrap(~t_sp1, scales="free_x")+
    facet_grid( .~ t_sp, scales = "free", space = "free") + 
    theme(legend.key.size = unit(.2, "cm")) + 
    guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
    #scale_fill_discrete(name="t_sp")+
    ggtitle("")+ ylab("Relative Abundance")+xlab("")
  ps.by_phylum_plot 
  ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/MPS.merged_by_phylum_0.005.pdf", ps.by_phylum_plot)
#
# Same but with Genus
ps.any.tr <- transform_sample_counts(merged_by_ss.genus, function (x) x / sum(x))
keep_percent<-.005
ps.any.f <- filter_taxa(ps.any.tr, function (x) mean(x) > keep_percent, TRUE) 
  patho.by_any_plot <- plot_bar(ps.any.f, "Sample", "Abundance", 'Genus', labs(y="Relative Abundance", title = "Genus-level assignments"))
  ps.by_genus_plot <- patho.by_any_plot + 
    theme_classic() + 
    #facet_wrap(~t_sp1, scales="free_x")+
    facet_grid( .~ t_sp, scales = "free", space = "free") + 
    theme(legend.key.size = unit(.2, "cm")) + 
    guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
    ylab("Relative Abundance")+xlab("")+
    #scale_fill_discrete(name="t_sp")+
    ggtitle("")
  ps.by_genus_plot 
  ggsave("~/Box/termite_microbiome/Dada2/Jan2022/plots/MPS.merged_by_genus_0.005.pdf", ps.by_phylum_plot)
  multiplot(ps.by_phylum_plot,ps.by_genus_plot)
```
Bar plot with common Pathways?
```{r}
#Combine all the ones that start with PWY

# add a label here that includes both phylum and genus
myranks = c("Phylum", "Genus")
Phylum_Genus = apply(tax_table(ps_arch)[, myranks], 1, paste, sep="", collapse=": ")
# Add concatenated labels as a new rank after strain
tax_table(ps_arch) <- cbind(tax_table(ps_arch), catglab=Phylum_Genus)


ps.any.tr <- transform_sample_counts(ps9, function (x) x / sum(x))
keep_percent<-.008
ps.any.f <- filter_taxa(ps.any.tr, function (x) mean(x) > keep_percent, TRUE) 
  patho.by_any_plot <- plot_bar(ps.any.f, "Sample", "Abundance","catglab")
  ps.by_fun_plot <- patho.by_any_plot + 
    theme_classic() + 
    #facet_wrap(~t_sp1, scales="free_x")+
    facet_grid( .~ t_sp, scales = "free", space = "free") + 
    theme(legend.key.size = unit(.2, "cm")) + 
    guides(fill=guide_legend(ncol=1,byrow=TRUE)) +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+
    ylab("Relative Abundance")+xlab("")+
    #scale_fill_discrete(name="t_sp")+
    ggtitle("Pathways")
  ps.by_fun_plot 
```

Next analyses: Look specifically at the archaea to see what's happening there.
```{r}
ps_arch<-subset_taxa(ps.bygenus.tr, Kingdom=="Archaea" )
# add a label here that includes both phylum and genus
myranks = c("Phylum", "Genus")
Phylum_Genus = apply(tax_table(ps_arch)[, myranks], 1, paste, sep="", collapse=": ")
# Add concatenated labels as a new rank after strain
tax_table(ps_arch) <- cbind(tax_table(ps_arch), catglab=Phylum_Genus)
#add the sp_habitat category and merge samples
ps_arch@sam_data$sp_habitat<-paste(ps_arch@sam_data$t_sp,ps_arch@sam_data$Site2)
merged_by_ss.archaea = merge_samples(ps_arch, "sp_habitat")


arch_plot<-plot_bar(merged_by_ss.archaea,"Sample", "Abundance",fill="catglab")
arch_plot+ylab("Relative Abundance")+xlab("")+facet_grid( .~ t_sp1, scales = "free", space = "free")+theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+labs(fill="Phylum: Genus")
```
Planctomycetota
```{r}
```{r}
ps_planct<-subset_taxa(ps.bygenus.tr, Phylum=="Planctomycetota" )
# add a label here that includes both phylum and genus
myranks = c("Family", "Genus")
Family_Genus = apply(tax_table(ps_planct)[, myranks], 1, paste, sep="", collapse=": ")
# Add concatenated labels as a new rank after strain
tax_table(ps_planct) <- cbind(tax_table(ps_planct), catglab=Family_Genus)
#add the sp_habitat category and merge samples
ps_planct@sam_data$sp_habitat<-paste(ps_planct@sam_data$t_sp,ps_planct@sam_data$Site2)
merged_by_ss.planct = merge_samples(ps_planct, "sp_habitat")


planct_plot<-plot_bar(merged_by_ss.planct,"Sample", "Abundance",fill="catglab")
planct_plot+ylab("Relative Abundance")+xlab("")+facet_grid( .~ t_sp1, scales = "free", space = "free")+theme_classic() +
    theme(axis.text.x = element_text(angle = 90, vjust=0.5, hjust=0))+labs(fill="Family: Genus")
```


```{r}
# heat map
ps.bygenus.tr.f@sam_data$sp_habitat<-paste(ps.bygenus.tr.f@sam_data$t_sp,ps.bygenus.tr.f@sam_data$Site2)
#sampleOrder = unique(sample_names(ps.bygenus.tr.f))
# sort taxa
#devtools::install_github("david-barnett/microViz@0.9.0")
#library(microViz)
ps.bygenus.tr.f %>%
  tax_sort(by = "name", at = "Phylum") %>%
  tax_table() %>%
  head(30)
taxaOrder = unique(taxa_names(ps.bygenus.tr.f[order(ps.bygenus.tr.f@tax_table),]))
plot_heatmap(ps.bygenus.tr.f, sample.label="sp_habitat", sample.order="t_sp", taxa.label = "catglab",low="#66CCFF",high="#000033",taxa.order=taxaOrder)

```
# make cladogram of similarities
```{r}
#https://userweb.eng.gla.ac.uk/umer.ijaz/bioinformatics/ecological.html
library(ggdendro)
 
#Transpose the data to have sample names on rows
abund_table<-otu_table(merged_by_ss.phy)
 
#Get grouping information
#grouping_info<-data.frame(row.names=rownames(abund_table),t(as.data.frame(strsplit(rownames(abund_table),"_"))))
grouping_info<-as.data.frame(merged_by_ss.phy@sam_data)
#grouping_info$sp_habitat<-as.factor(grouping_info$sp_habitat)
grouping_info$host_genus<-ifelse(grouping_info$t_sp1==1,"Amitermes",ifelse(grouping_info$t_sp1==2,"Microcerotermes",ifelse(grouping_info$t_sp1==3,"Nasutitermes","Coptotermes")))

betad<-vegdist(abund_table,method="bray")
 
# Use Adonis to test for overall differences
res_adonis <- adonis(betad ~ t_sp, grouping_info) 
 
#Cluster the samples
hc <- hclust(betad)
 
#We will color the labels according to countries(group_info[,1])
hc_d <- dendro_data(as.dendrogram(hc))
hc_d$labels$Type<-grouping_info[as.character(hc_d$labels$label),18]
hc_d$labels$Type<-hc_d$labels$Type$host_genus
#Coloring function
gg_color_hue<-function(n){
  hues=seq(15,375,length=n+1)
  hcl(h=hues,l=65,c=100)[1:n]
}
 
cols=gg_color_hue(length(unique(hc_d$labels$Type)))
#hc_d$labels$color=cols[hc_d$labels$Type]
hc_d$labels$color=c(cols[1],cols[1],cols[2],cols[2],cols[2],cols[2],cols[2],cols[3],cols[3],cols[4],cols[4])

## Plot clusters
p1 <- ggplot(data = segment(hc_d)) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  coord_flip() +
  scale_x_discrete(labels=label(hc_d)$label) +
  ylab("Distance (beta diversity = bray)") + theme_bw()+
  theme(axis.text.y = element_text(color = hc_d$labels$color),
        axis.title.y = element_blank())
#p1+geom_text(aes(x = x, y = y, label = label, angle = -90, hjust = 0), data= hc_d$label) +
 # scale_y_continuous(expand = c(0.3, 0))
p1 <- p1 + geom_point(data=hc_d$label, aes(x = x, y = y, color = Type), size=3,inherit.aes =F, alpha = 0)
p1
p1 <- p1 + guides(colour = guide_legend(override.aes = list(size=5, alpha = 1)))+
  scale_color_manual(values = cols)
p1
pdf("Cluster.pdf",height=10)
print(p1)
dev.off()
ggdendrogram(hc_d,rotate=T)+ geom_text(data=hc_d$labels,
               aes(label(hc_d)$label, x=x, y=0, hc_d$labels$color))
labs<-label(hc_d)
p2 <- ggplot(segment(hc_d)) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend))
p2 + geom_text(data=label(hc_d),
               aes(label=label, x=x, y=0, colour=Type,size=.1))+coord_flip() +ylab("Distance (beta diversity = bray)") + theme_bw()+ geom_point(data=hc_d$label, aes(x = x, y = y, color = Type), inherit.aes =T, alpha = 0)
```
Same but without the merged samples
```{r}
#Transpose the data to have sample names on rows
abund_table<-t(otu_table(ps8))
 
#Get grouping information
#grouping_info<-data.frame(row.names=rownames(abund_table),t(as.data.frame(strsplit(rownames(abund_table),"_"))))
grouping_info<-as.data.frame(ps8@sam_data)
grouping_info$sp_habitat<-as.factor(grouping_info$sp_habitat)
betad<-vegdist(abund_table,method="bray")
#Cluster the samples
hc <- hclust(betad)
#We will color the labels according to countries(group_info[,1])
hc_d <- dendro_data(as.dendrogram(hc))
hc_d$labels$Type<-grouping_info[as.character(hc_d$labels$label),17]
hc_d$labels$Type<-hc_d$labels$Type$sp_habitat
#Coloring function
gg_color_hue<-function(n){
  hues=seq(15,375,length=n+1)
  hcl(h=hues,l=65,c=100)[1:n]
}
 
cols=gg_color_hue(length(unique(hc_d$labels$Type)))
hc_d$labels$color=cols[hc_d$labels$Type]

## Plot clusters
p1 <- ggplot(data = segment(hc_d)) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend)) +
  coord_flip() +
  scale_x_discrete(labels=label(hc_d)$label) +
  ylab("Distance (beta diversity = bray)") + theme_bw()+
  theme(axis.text.y = element_text(color = hc_d$labels$color),
        axis.title.y = element_blank())
#p1+geom_text(aes(x = x, y = y, label = label, angle = -90, hjust = 0), data= hc_d$label) +
 # scale_y_continuous(expand = c(0.3, 0))
p1 <- p1 + geom_point(data=hc_d$label, aes(x = x, y = y, color = Type), size=3,inherit.aes =F, alpha = 0)
p1
p1 <- p1 + guides(colour = guide_legend(override.aes = list(size=5, alpha = 1)))+
  scale_color_manual(values = cols)
p1
pdf("Cluster.pdf",height=10)
print(p1)
dev.off()
ggdendrogram(hc_d,rotate=T)+ geom_text(data=hc_d$labels,
               aes(label(hc_d)$label, x=x, y=0, label(hc_d)$color))
labs<-label(hc_d)
p2 <- ggplot(segment(hc_d)) +
  geom_segment(aes(x=x, y=y, xend=xend, yend=yend))
p2 + geom_text(data=label(hc_d),
               aes(label=label, x=x, y=0, colour=Type,size=.01))+coord_flip() +ylab("Distance (beta diversity = bray)") + theme_bw()+ geom_point(data=hc_d$label, aes(x = x, y = y, color = Type), inherit.aes =T, alpha = 0)
```
Next steps: We used species accumulation curves to see the additive effect of sampling across distance for savanna sites. We used autocorrelelograms?

Fig. Species accumulation curve with distance on x axis?
Fig. autocorrelelogram which shows how many meters becomes a different thing?
Fig 3.  Photos of the termites:)
https://cran.r-project.org/web/packages/acnr/vignettes/autocorrelation.html

Which species had the greatest betadiversity within site? 